---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%",
  warning = FALSE
)
```

# Sigminer: Mutational Signature Analysis and Visualization in R <img src="man/figures/logo.png" alt="logo" align="right" height="140" width="120"/>

[![CRAN status](https://www.r-pkg.org/badges/version/sigminer)](https://cran.r-project.org/package=sigminer) [![lifecycle](https://img.shields.io/badge/lifecycle-stable-blue.svg)](https://lifecycle.r-lib.org/articles/stages.html) [![R-CMD-check](https://github.com/ShixiangWang/sigminer/workflows/R-CMD-check/badge.svg)](https://github.com/ShixiangWang/sigminer/actions) [![Coverage status](https://codecov.io/gh/ShixiangWang/sigminer/branch/master/graph/badge.svg)](https://codecov.io/github/ShixiangWang/sigminer?branch=master) [![](https://cranlogs.r-pkg.org/badges/grand-total/sigminer?color=orange)](https://cran.r-project.org/package=sigminer) [![Closed issues](https://img.shields.io/github/issues-closed/ShixiangWang/sigminer.svg)](https://github.com/ShixiangWang/sigminer/issues?q=is%3Aissue+is%3Aclosed) [![Lines Of Code](https://tokei.rs/b1/github/ShixiangWang/sigminer?category=code)](https://github.com/ShixiangWang/sigminer) [![Hits](https://hits.seeyoufarm.com/api/count/incr/badge.svg?url=https%3A%2F%2Fgithub.com%2FShixiangWang%2Fsigminer&count_bg=%2379C83D&title_bg=%23555555&icon=&icon_color=%23E7E7E7&title=hits&edge_flat=false)](https://hits.seeyoufarm.com) ![install with bioconda](https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat-square) [![check in Biotreasury](https://img.shields.io/badge/Biotreasury-collected-brightgreen)](https://biotreasury.rjmart.cn/#/tool?id=10043) 


## :bar_chart: Overview

The cancer genome is shaped by various mutational processes over its lifetime, stemming from exogenous and cell-intrinsic DNA damage, and error-prone DNA replication, leaving behind characteristic mutational spectra, termed **mutational signatures**. This package, **sigminer**, helps users to extract, analyze and visualize signatures from genome alteration records, thus providing new insight into cancer study.

For pipeline tool, please see its co-evolutionary CLI [sigflow](https://github.com/ShixiangWang/sigflow).

**SBS signatures**:

```{r, fig.width=12, fig.height=5, echo=FALSE, message=FALSE}
library(sigminer)
load(system.file("extdata", "toy_mutational_signature.RData",
  package = "sigminer", mustWork = TRUE
))
# Show signature profile
p1 <- show_sig_profile(sig2, mode = "SBS", style = "cosmic", x_label_angle = 90)
p1

```

**Copy number signatures**:

```{r fig.width=12, fig.height=5, echo=FALSE}
load(system.file("extdata", "toy_copynumber_signature_by_W.RData",
  package = "sigminer", mustWork = TRUE
))
# Show signature profile
p2 <- show_sig_profile(sig,
  style = "cosmic",
  mode = "copynumber",
  method = "W",
  normalize = "feature"
)
p2
```

**DBS signatures**:

```{r fig.width=12, fig.height=5, echo=FALSE}
DBS = system.file("extdata", "DBS_signatures.rds",
                        package = "sigminer", mustWork = TRUE)
DBS = readRDS(DBS)

# Show signature profile
p3 <- show_sig_profile(DBS$db[, 1:3] %>% as.matrix(), mode = "DBS", style = "cosmic", check_sig_names = F)
p3
```

**INDEL (i.e. ID) signatures**:

```{r fig.width=12, fig.height=5, echo=FALSE}
ID = system.file("extdata", "ID_signatures.rds",
                        package = "sigminer", mustWork = TRUE)
ID = readRDS(ID)

# Show signature profile
p4 <- show_sig_profile(ID$db[, 4:6] %>% as.matrix(), mode = "ID", style = "cosmic", check_sig_names = F)
p4
```

**Genome rearrangement signatures**:

```{r fig.width=10, fig.height=5, echo=FALSE, message=FALSE}
p5 <- show_cosmic_sig_profile(sig_index = c("R1", "R2", "R3"), sig_db = "RS_Nik_lab", style = "cosmic", show_index = FALSE)
p5
```


### :airplane: Features

-   supports a standard *de novo* pipeline for identification of **5** types of signatures: copy number, SBS, DBS, INDEL and RS (genome rearrangement signature).
-   supports quantify exposure for one sample based on *known signatures*.
-   supports association and group analysis and visualization for signatures.
-   supports two types of signature exposures: relative exposure (relative contribution of signatures in each sample) and absolute exposure (estimated variation records of signatures in each sample).
-   supports basic summary and visualization for profile of mutation (powered by **maftools**) and copy number.
-   supports parallel computation by R packages **foreach**, **future** and **NMF**.
-   efficient code powered by R packages **data.table** and **tidyverse**.
-   elegant plots powered by R packages **ggplot2**, **ggpubr**, **cowplot** and **patchwork**.
-   well tested by R package **testthat** and documented by R package **roxygen2**, **roxytest**, **pkgdown**, and etc. for both reliable and reproducible research.


## :arrow_double_down: Installation

You can install the stable release of **sigminer** from CRAN with:

```{r, eval=FALSE}
install.packages("BiocManager")
BiocManager::install("sigminer", dependencies = TRUE)
```

You can install the development version of **sigminer** from Github with:

```{r, eval=FALSE}
remotes::install_github("ShixiangWang/sigminer", dependencies = TRUE)
# For Chinese users, run 
remotes::install_git("https://gitee.com/ShixiangWang/sigminer", dependencies = TRUE)
```

You can also install **sigminer** from conda `bioconda` channel with

```sh
# Please note version number of the bioconda release

# You can install an individual environment firstly with
# conda create -n sigminer
# conda activate sigminer
conda install -c bioconda -c conda-forge r-sigminer
```

## :beginner: Usage

A complete documentation of **sigminer** can be read online at <https://shixiangwang.github.io/sigminer-book/>. All functions are well organized and documented at <https://shixiangwang.github.io/sigminer/reference/index.html>. For usage of a specific function `fun`, run `?fun` in your R console to see its documentation.

## :paperclip: Citation

If you use **sigminer** in academic field, please cite one of the following papers.

------------------------------------------------------------------------

-   ***Wang S, Li H, Song M, Tao Z, Wu T, He Z, et al. (2021) Copy number signature analysis tool and its application in prostate cancer reveals distinct mutational processes and clinical outcomes. PLoS Genet 17(5): e1009557.*** <https://doi.org/10.1371/journal.pgen.1009557>

-   ***Shixiang Wang, Ziyu Tao, Tao Wu, Xue-Song Liu, Sigflow: An Automated And Comprehensive Pipeline For Cancer Genome Mutational Signature Analysis, Bioinformatics, btaa895***. <https://doi.org/10.1093/bioinformatics/btaa895>

------------------------------------------------------------------------

## :arrow_down: Download Stats

```{r, echo=FALSE, warning=FALSE, message=FALSE}
library("ggplot2")
library("cranlogs")
library("dplyr")

x <- cran_downloads("sigminer", from = "2019-07-01", to = "2021-09-01")

if (!is.null(x)) {
   #head(x)
   p <- ggplot(x, aes(date, cumsum(count), group=package, color=package)) +
       geom_line() + geom_point(aes(shape=package)) + 
    theme_minimal() + theme(legend.position = "none") +
    labs(x = "Time", y = "Cumulative downloads")
   p
}
```

## :page_with_curl: References

Please properly cite the following references when you are using any corresponding features. The references are also listed in the function documentation. Very thanks to the works, **sigminer** cannot be created without the giants.

1.  Mayakonda, Anand, et al. "Maftools: efficient and comprehensive analysis of somatic variants in cancer." Genome research 28.11 (2018): 1747-1756.
2.  Gaujoux, Renaud, and Cathal Seoighe. "A Flexible R Package for Nonnegative Matrix Factorization."" BMC Bioinformatics 11, no. 1 (December 2010).
3.  H. Wickham. ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York, 2016.
4.  Kim, Jaegil, et al. "Somatic ERCC2 mutations are associated with a distinct genomic signature in urothelial tumors." Nature genetics 48.6 (2016): 600.
5.  Alexandrov, Ludmil B., et al. "Deciphering signatures of mutational processes operative in human cancer." Cell reports 3.1 (2013): 246-259.
6.  Degasperi, Andrea, et al. "A practical framework and online tool for mutational signature analyses show intertissue variation and driver dependencies." Nature cancer 1.2 (2020): 249-263.
7.  Alexandrov, Ludmil B., et al. "The repertoire of mutational signatures in human cancer." Nature 578.7793 (2020): 94-101.
8.  Macintyre, Geoff, et al. "Copy number signatures and mutational processes in ovarian carcinoma." Nature genetics 50.9 (2018): 1262.
9.  Tan, Vincent YF, and Cédric Févotte. "Automatic relevance determination in nonnegative matrix factorization with the/spl beta/-divergence." IEEE Transactions on Pattern Analysis and Machine Intelligence 35.7 (2012): 1592-1605.
10. Bergstrom EN, Huang MN, Mahto U, Barnes M, Stratton MR, Rozen SG, Alexandrov LB: SigProfilerMatrixGenerator: a tool for visualizing and exploring patterns of small mutational events. BMC Genomics 2019, 20:685 <https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6041-2>

## :page_facing_up: LICENSE

The software is made available for non commercial research purposes only under the [MIT](https://github.com/ShixiangWang/sigminer/blob/master/LICENSE.md). However, notwithstanding any provision of the MIT License, the software currently may not be used for commercial purposes without explicit written permission after contacting Shixiang Wang [wangshx\@shanghaitech.edu.cn](mailto:wangshx@shanghaitech.edu.cn) or Xue-Song Liu [liuxs\@shanghaitech.edu.cn](mailto:liuxs@shanghaitech.edu.cn).

MIT © 2019-Present Shixiang Wang, Xue-Song Liu

MIT © 2018 Anand Mayakonda

------------------------------------------------------------------------

[**Cancer Biology Group**](https://github.com/XSLiuLab) **\@ShanghaiTech**

**Research group led by Xue-Song Liu in ShanghaiTech University**

![Alt](https://repobeats.axiom.co/api/embed/7cd2cf8a196dde9d8d1e13c9b23bc2f157d8254e.svg "Repobeats analytics image")
