<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • sigminer</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><meta property="og:image" content="https://shixiangwang.github.io/sigminer/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sigminer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/sigminer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cnsignature.html">Analyze Copy Number Signatures with sigminer</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ShixiangWang/sigminer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/ShixiangWang/sigminer/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="2.3.0" id="sigminer-230">sigminer 2.3.0<a class="anchor" aria-label="anchor" href="#sigminer-230"></a></h2>
<ul><li>Supported human T2T genome and corresponding annotation data.</li>
<li>Updated COSMIC database to v3.4. SV and RNA-SBS signatures are included.</li>
</ul><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/get_sig_db.html">get_sig_db</a></span><span class="op">(</span><span class="st">"latest_RNA-SBS_GRCh37"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/get_sig_db.html">get_sig_db</a></span><span class="op">(</span><span class="st">"latest_SV_GRCh38"</span><span class="op">)</span></span></code></pre></div>
<ul><li>Fixed a bug in generating matrix for variation categories with strand bias due to problematic counting. (<a href="https://github.com/ShixiangWang/sigminer/issues/445" class="external-link">#445</a>)</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.2.2" id="sigminer-222">sigminer 2.2.2<small>2023-08-21</small><a class="anchor" aria-label="anchor" href="#sigminer-222"></a></h2>
<ul><li>Updated pkg doc following the new CRAN feature (thanks to K from the CRAN team).</li>
<li>Added <code>samps</code> option to <code><a href="../reference/show_sig_exposure.html">show_sig_exposure()</a></code>.</li>
</ul><p>Example:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/R.oo/man/Non-documented_objects.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"toy_mutational_signature.RData"</span>,</span>
<span>                 package <span class="op">=</span> <span class="st">"sigminer"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Show signature exposure</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/show_sig_exposure.html">show_sig_exposure</a></span><span class="op">(</span><span class="va">sig2</span>, rm_space <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p1</span></span>
<span></span>
<span><span class="va">expo</span> <span class="op">=</span> <span class="fu"><a href="../reference/sig_operation.html">sig_exposure</a></span><span class="op">(</span><span class="va">sig2</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/show_sig_exposure.html">show_sig_exposure</a></span><span class="op">(</span><span class="va">expo</span>,</span>
<span>                  rm_space <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  samps <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">expo</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">expo</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.2.1" id="sigminer-221">sigminer 2.2.1<a class="anchor" aria-label="anchor" href="#sigminer-221"></a></h2>
<ul><li>Fixed the error in generating SBS matrix when only one sample input (<a href="https://github.com/ShixiangWang/sigminer/issues/432" class="external-link">#432</a>).</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.2.0" id="sigminer-220">sigminer 2.2.0<small>2023-04-06</small><a class="anchor" aria-label="anchor" href="#sigminer-220"></a></h2>
<ul><li>Removed package ‘copynumber’ from suggests filed.</li>
<li>Supported Ziyu Tao et al approach for copy number segment classification.</li>
<li>Supported ce11 genome in <code><a href="../reference/read_vcf.html">read_vcf()</a></code>.</li>
<li>Added <code><a href="../reference/read_maf.html">read_maf_minimal()</a></code> to support a minimal MAF-like data as input.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.10" id="sigminer-2110">sigminer 2.1.10<a class="anchor" aria-label="anchor" href="#sigminer-2110"></a></h2>
<ul><li>Fixed the issue about the latest CN signatures from COSMIC have inconsistent labels with built-in CN signatures (<a href="https://github.com/ShixiangWang/sigminer/issues/421" class="external-link">#421</a>).</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.9" id="sigminer-219">sigminer 2.1.9<small>2022-11-09</small><a class="anchor" aria-label="anchor" href="#sigminer-219"></a></h2>
<ul><li>Sorted substitution mutation types by default in <code><a href="../reference/sig_tally.html">sig_tally()</a></code>.</li>
<li>Added parameter in <code><a href="../reference/sigprofiler.html">sigprofiler_extract()</a></code> to help generate input matrix file for calling SigProfiler directly.</li>
<li>Added some notions in <code><a href="../reference/sigprofiler.html">sigprofiler_extract()</a></code>.</li>
<li>Added a function <code><a href="../reference/sigprofiler.html">sigprofiler_reorder()</a></code> for utils in generating SigProfiler input matrix file with standard mutation types order.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.8" id="sigminer-218">sigminer 2.1.8<small>2022-10-20</small><a class="anchor" aria-label="anchor" href="#sigminer-218"></a></h2>
<ul><li>Fixed the bug about plotting CN chromosome distribution (<a href="https://github.com/ShixiangWang/sigminer/issues/420" class="external-link">#420</a>, thanks to <a href="https://github.com/jrcodina96" class="external-link">@jrcodina96</a>).</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.7" id="sigminer-217">sigminer 2.1.7<small>2022-08-31</small><a class="anchor" aria-label="anchor" href="#sigminer-217"></a></h2>
<ul><li>Updated COSMIC latest version from v3.2 to v3.3. A new reference for copy number signature now is provided as <code>latest_CN_GRCh37</code> (<a href="https://github.com/ShixiangWang/sigminer/issues/412" class="external-link">#412</a>).</li>
<li>
<code><a href="../reference/get_sig_similarity.html">get_sig_similarity()</a></code> now uses “SBS” as default reference.</li>
<li>Fixed bug in <code><a href="../reference/show_cn_circos.html">show_cn_circos()</a></code>.</li>
<li>Added <code><a href="../reference/group_enrichment2.html">group_enrichment2()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.6" id="sigminer-216">sigminer 2.1.6<small>2022-08-09</small><a class="anchor" aria-label="anchor" href="#sigminer-216"></a></h2>
<ul><li>Fixed checking in <code><a href="../reference/sig_tally.html">sig_tally()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.5" id="sigminer-215">sigminer 2.1.5<small>2022-06-30</small><a class="anchor" aria-label="anchor" href="#sigminer-215"></a></h2>
<ul><li>Added a vignette to introduce the analysis of copy number signatures.</li>
<li>Updated <code>CNS_TCGA</code>.</li>
<li>Enhanced <code><a href="../reference/group_enrichment.html">group_enrichment()</a></code> with reference group support.</li>
</ul><p>Example:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>  g1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">40</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  g2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AA"</span>, <span class="st">"VV"</span>, <span class="st">"XX"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">40</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  e1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"P"</span>, <span class="st">"N"</span><span class="op">)</span>, <span class="fl">100</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  e2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">x1</span> <span class="op">=</span> <span class="fu"><a href="../reference/group_enrichment.html">group_enrichment</a></span><span class="op">(</span><span class="va">df</span>, grp_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"g1"</span>, <span class="st">"g2"</span><span class="op">)</span>, </span>
<span>                      enrich_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"e1"</span>, <span class="st">"e2"</span><span class="op">)</span>, </span>
<span>                      ref_group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B"</span>, <span class="st">"VV"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x1</span></span></code></pre></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.4" id="sigminer-214">sigminer 2.1.4<small>2022-04-26</small><a class="anchor" aria-label="anchor" href="#sigminer-214"></a></h2>
<ul><li>Added option for reading ASCAT objects in parallel.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.3" id="sigminer-213">sigminer 2.1.3<small>2022-03-10</small><a class="anchor" aria-label="anchor" href="#sigminer-213"></a></h2>
<ul><li>Fixed error in extracting invalid regions (<a href="https://github.com/ShixiangWang/sigminer/issues/396" class="external-link">#396</a>, thanks to <a href="https://github.com/KirsieMin" class="external-link">@KirsieMin</a>).</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.2" id="sigminer-212">sigminer 2.1.2<small>2021-12-15</small><a class="anchor" aria-label="anchor" href="#sigminer-212"></a></h2>
<ul><li>Enhanced the <code><a href="../reference/read_copynumber_seqz.html">read_copynumber_seqz()</a></code> to include minor copy number. (Thanks to yancey)</li>
<li>Added input <code>range</code> check in <code><a href="../reference/sig_estimate.html">sig_estimate()</a></code>. (<a href="https://github.com/ShixiangWang/sigminer/issues/391" class="external-link">#391</a>)</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.1" id="sigminer-211">sigminer 2.1.1<small>2021-10-29</small><a class="anchor" aria-label="anchor" href="#sigminer-211"></a></h2>
<ul><li>Expanded <code>output_*</code> function by adding option <code>sig_db</code>.</li>
<li>Fixed the error using <code><a href="../reference/get_genome_annotation.html">sigminer::get_genome_annotation()</a></code> before loading it.</li>
<li>Fixed the bug the <code><a href="../reference/get_pLOH_score.html">get_pLOH_score()</a></code> return nothing for sample without LOH.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.0" id="sigminer-210">sigminer 2.1.0<small>2021-09-22</small><a class="anchor" aria-label="anchor" href="#sigminer-210"></a></h2>
<ul><li>Added <code><a href="../reference/sig_unify_extract.html">sig_unify_extract()</a></code> as an unified signature extractor.</li>
<li>Fixed error showing reference signature profile for <code>CNS_TCGA</code> database.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.5" id="sigminer-205">sigminer 2.0.5<small>2021-09-03</small><a class="anchor" aria-label="anchor" href="#sigminer-205"></a></h2>
<ul><li>Impl <code>y_limits</code> option in <code><a href="../reference/show_sig_profile.html">show_sig_profile()</a></code> (<a href="https://github.com/ShixiangWang/sigminer/issues/381" class="external-link">#381</a>).</li>
<li>Added function <code><a href="../reference/get_pLOH_score.html">get_pLOH_score()</a></code> for representing the genome that displayed LOH.</li>
<li>Added function <code><a href="../reference/read_copynumber_ascat.html">read_copynumber_ascat()</a></code> for reading ASCAT result ASCAT object in <code>.rds</code> format.</li>
<li>Added function <code><a href="../reference/get_intersect_size.html">get_intersect_size()</a></code> for getting overlap size between intervals.</li>
<li>Added option to <code><a href="../reference/get_Aneuploidy_score.html">get_Aneuploidy_score()</a></code> to remove short arms of chr13/14/15/21/22 from calculation.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.4" id="sigminer-204">sigminer 2.0.4<small>2021-08-03</small><a class="anchor" aria-label="anchor" href="#sigminer-204"></a></h2>
<ul><li>Implemented Cohen-Sharir method-like Aneuploidy Score.</li>
<li>Enhanced error handling in <code><a href="../reference/show_sig_feature_corrplot.html">show_sig_feature_corrplot()</a></code> (<a href="https://github.com/ShixiangWang/sigminer/issues/376" class="external-link">#376</a>).</li>
<li>Fixed INDEL classification.</li>
<li>Fixed end position determination in <code><a href="../reference/read_vcf.html">read_vcf()</a></code>.</li>
<li>Updated INDEL adjustment.</li>
<li>Included TCGA copy number signatures from SigProfiler.</li>
<li>Updated docs.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.3" id="sigminer-203">sigminer 2.0.3<small>2021-07-18</small><a class="anchor" aria-label="anchor" href="#sigminer-203"></a></h2>
<ul><li>Preprocessed INDELs before labeling them in <code><a href="../reference/sig_tally.html">sig_tally()</a></code> (<a href="https://github.com/ShixiangWang/sigminer/issues/370" class="external-link">#370</a>).</li>
<li>Fixed <code><a href="../reference/sigprofiler.html">sigprofiler_extract()</a></code> extracting copy number signatures and rolled up sigprofiler version (<a href="https://github.com/ShixiangWang/sigminer/issues/369" class="external-link">#369</a>).</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.2" id="sigminer-202">sigminer 2.0.2<small>2021-06-17</small><a class="anchor" aria-label="anchor" href="#sigminer-202"></a></h2>
<ul><li>Fixed <code><a href="../reference/output_sig.html">output_sig()</a></code> error in handling exposure plot with &gt;9 signatures (<a href="https://github.com/ShixiangWang/sigminer/issues/366" class="external-link">#366</a>).</li>
<li>Added <code>limitsize = FALSE</code> for <code><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave()</a></code> or <code>ggsave2()</code> for handling big figure.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.1" id="sigminer-201">sigminer 2.0.1<small>2021-05-19</small><a class="anchor" aria-label="anchor" href="#sigminer-201"></a></h2>
<ul><li>Supported <code>mm9</code> genome build.</li>
<li>Removed FTP link as CRAN suggested (<a href="https://github.com/ShixiangWang/sigminer/issues/359" class="external-link">#359</a>).</li>
<li>Updated README.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.0" id="sigminer-200">sigminer 2.0.0<small>2021-04-01</small><a class="anchor" aria-label="anchor" href="#sigminer-200"></a></h2>
<div class="section level3">
<h3 id="bug-reports-2-0-0">BUG REPORTS<a class="anchor" aria-label="anchor" href="#bug-reports-2-0-0"></a></h3>
<ul><li>Fixed the SigProfiler installation error due to Python version in conda environment.</li>
<li>Fixed classification bug due to repeated function name <code>call_component</code>.</li>
<li>Fixed the bug when <code><a href="../reference/read_vcf.html">read_vcf()</a></code> with <code>##</code> commented VCF files.</li>
</ul></div>
<div class="section level3">
<h3 id="enhancements-2-0-0">ENHANCEMENTS<a class="anchor" aria-label="anchor" href="#enhancements-2-0-0"></a></h3>
<ul><li>Added support for latest COSMIC v3.2 as reference signatures. You can obtain them by</li>
</ul><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"latest_SBS_GRCh37"</span>, <span class="st">"latest_DBS_GRCh37"</span>, <span class="st">"latest_ID_GRCh37"</span>,</span>
<span>            <span class="st">"latest_SBS_GRCh38"</span>, <span class="st">"latest_DBS_GRCh38"</span>,</span>
<span>            <span class="st">"latest_SBS_mm9"</span>, <span class="st">"latest_DBS_mm9"</span>,</span>
<span>            <span class="st">"latest_SBS_mm10"</span>, <span class="st">"latest_DBS_mm10"</span>,</span>
<span>            <span class="st">"latest_SBS_rn6"</span>, <span class="st">"latest_DBS_rn6"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="../reference/get_sig_db.html">get_sig_db</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<ul><li>Updated <code>keep_only_pass</code> to <code>FALSE</code> at default.</li>
<li>Added RSS and unexplained variance calculation in <code><a href="../reference/get_sig_rec_similarity.html">get_sig_rec_similarity()</a></code>.</li>
<li>Added data check and filter in <code><a href="../reference/output_tally.html">output_tally()</a></code> and <code><a href="../reference/show_catalogue.html">show_catalogue()</a></code>.</li>
<li>Enhanced <code><a href="../reference/show_group_enrichment.html">show_group_enrichment()</a></code> (<a href="https://github.com/ShixiangWang/sigminer/issues/353" class="external-link">#353</a>) &amp; added a new option to cluster rows.</li>
<li>Removed unnecessary CN classifications code in recent development.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-2-0-0">NEW FUNCTIONS<a class="anchor" aria-label="anchor" href="#new-functions-2-0-0"></a></h3>
</div>
<div class="section level3">
<h3 id="deprecated-2-0-0">DEPRECATED<a class="anchor" aria-label="anchor" href="#deprecated-2-0-0"></a></h3>
<ul><li>Dropped copy number “M”” method to avoid misguiding user to use/read wrong signature profile and keep code simple.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.2.5" id="sigminer-125">sigminer 1.2.5<small>2021-02-20</small><a class="anchor" aria-label="anchor" href="#sigminer-125"></a></h2>
<div class="section level3">
<h3 id="bug-reports-1-2-5">BUG REPORTS<a class="anchor" aria-label="anchor" href="#bug-reports-1-2-5"></a></h3>
</div>
<div class="section level3">
<h3 id="enhancements-1-2-5">ENHANCEMENTS<a class="anchor" aria-label="anchor" href="#enhancements-1-2-5"></a></h3>
<ul><li>Modified the default visualization of <code><a href="../reference/bp.html">bp_show_survey()</a></code>.</li>
<li>Enhanced <code>torch</code> check.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-1-2-5">NEW FUNCTIONS<a class="anchor" aria-label="anchor" href="#new-functions-1-2-5"></a></h3>
<ul><li>
<code><a href="../reference/read_sv_as_rs.html">read_sv_as_rs()</a></code> and <code><a href="../reference/sig_tally.html">sig_tally.RS()</a></code> for simplified genome rearrangement classification matrix generation (experimental).</li>
</ul></div>
<div class="section level3">
<h3 id="deprecated-1-2-5">DEPRECATED<a class="anchor" aria-label="anchor" href="#deprecated-1-2-5"></a></h3>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.2.4" id="sigminer-124">sigminer 1.2.4<small>2021-01-30</small><a class="anchor" aria-label="anchor" href="#sigminer-124"></a></h2>
<div class="section level3">
<h3 id="bug-reports-1-2-4">BUG REPORTS<a class="anchor" aria-label="anchor" href="#bug-reports-1-2-4"></a></h3>
<ul><li>Fixed the assign problem about match pair in <code><a href="../reference/bp.html">bp_extract_signatures()</a></code> with <code>lpSolve</code> package instead of using my problematic code.</li>
</ul></div>
<div class="section level3">
<h3 id="enhancements-1-2-4">ENHANCEMENTS<a class="anchor" aria-label="anchor" href="#enhancements-1-2-4"></a></h3>
<ul><li>Supported <code>mm10</code> in <code><a href="../reference/read_vcf.html">read_vcf()</a></code>.</li>
<li>Removed large data files and store them in Zenodo to reduce package size.</li>
<li>Added cores check.</li>
<li>Upgraded SP to v1.1.0 (need test).</li>
<li>Tried installing Torch before SP (need test).</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-1-2-4">NEW FUNCTIONS<a class="anchor" aria-label="anchor" href="#new-functions-1-2-4"></a></h3>
</div>
<div class="section level3">
<h3 id="deprecated-1-2-4">DEPRECATED<a class="anchor" aria-label="anchor" href="#deprecated-1-2-4"></a></h3>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.2.2" id="sigminer-122">sigminer 1.2.2<a class="anchor" aria-label="anchor" href="#sigminer-122"></a></h2>
<div class="section level3">
<h3 id="bug-reports-1-2-2">BUG REPORTS<a class="anchor" aria-label="anchor" href="#bug-reports-1-2-2"></a></h3>
<ul><li>Fixed bug in silhouette calculation in <code><a href="../reference/bp.html">bp_extract_signatures()</a></code> (<a href="https://github.com/ShixiangWang/sigminer/issues/332" class="external-link">#332</a>). PAY ATTENTION: this may affect results.</li>
<li>Fixed bug using custom signature name in <code><a href="../reference/show_sig_profile_loop.html">show_sig_profile_loop()</a></code>.</li>
</ul></div>
<div class="section level3">
<h3 id="enhancements-1-2-2">ENHANCEMENTS<a class="anchor" aria-label="anchor" href="#enhancements-1-2-2"></a></h3>
<ul><li>Subset signatures to plot is available by <code>sig_names</code> option.</li>
<li>sigminer is available in bioconda channel: <code>https://anaconda.org/bioconda/r-sigminer/</code>
</li>
<li>Updated <code>ms</code> strategy in <code><a href="../reference/sig_auto_extract.html">sig_auto_extract()</a></code> by assigning each signature to its best matched reference signatures.</li>
<li>Added <code><a href="../reference/get_shannon_diversity_index.html">get_shannon_diversity_index()</a></code> to get diversity index for signatures (<a href="https://github.com/ShixiangWang/sigminer/issues/333" class="external-link">#333</a>).</li>
<li>Added new method “S” (from Steele et al. 2019) for tallying copy number data (<a href="https://github.com/ShixiangWang/sigminer/issues/329" class="external-link">#329</a>).</li>
<li>Included new (RS) reference signatures (related to <a href="https://github.com/ShixiangWang/sigminer/issues/331" class="external-link">#331</a>).</li>
<li>Updated the internal code for getting relative activity in <code><a href="../reference/get_sig_exposure.html">get_sig_exposure()</a></code>.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-1-2-2">NEW FUNCTIONS<a class="anchor" aria-label="anchor" href="#new-functions-1-2-2"></a></h3>
<ul><li>
<code><a href="../reference/bp.html">bp_get_clustered_sigs()</a></code> to get clustered mean signatures.</li>
</ul></div>
<div class="section level3">
<h3 id="deprecated-1-2-2">DEPRECATED<a class="anchor" aria-label="anchor" href="#deprecated-1-2-2"></a></h3>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.2.1" id="sigminer-121">sigminer 1.2.1<small>2021-01-08</small><a class="anchor" aria-label="anchor" href="#sigminer-121"></a></h2>
<ul><li>Updated author list.</li>
</ul><div class="section level3">
<h3 id="bug-reports-1-2-1">BUG REPORTS<a class="anchor" aria-label="anchor" href="#bug-reports-1-2-1"></a></h3>
</div>
<div class="section level3">
<h3 id="enhancements-1-2-1">ENHANCEMENTS<a class="anchor" aria-label="anchor" href="#enhancements-1-2-1"></a></h3>
<ul><li>Added a quick start vignette.</li>
<li>A new option <code>highlight</code> is added to <code><a href="../reference/sig_estimate.html">show_sig_number_survey()</a></code> and <code><a href="../reference/bp.html">bp_show_survey2()</a></code> to highlight a selected number.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-1-2-1">NEW FUNCTIONS<a class="anchor" aria-label="anchor" href="#new-functions-1-2-1"></a></h3>
</div>
<div class="section level3">
<h3 id="deprecated-1-2-1">DEPRECATED<a class="anchor" aria-label="anchor" href="#deprecated-1-2-1"></a></h3>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.2.0" id="sigminer-120">sigminer 1.2.0<a class="anchor" aria-label="anchor" href="#sigminer-120"></a></h2>
<div class="section level3">
<h3 id="bug-reports-1-2-0">BUG REPORTS<a class="anchor" aria-label="anchor" href="#bug-reports-1-2-0"></a></h3>
</div>
<div class="section level3">
<h3 id="enhancements-1-2-0">ENHANCEMENTS<a class="anchor" aria-label="anchor" href="#enhancements-1-2-0"></a></h3>
<ul><li>A new option <code>cut_p_value</code> is added to <code><a href="../reference/show_group_enrichment.html">show_group_enrichment()</a></code> to cut continous p values as binned regions.</li>
<li>A Python backend for <code><a href="../reference/sig_extract.html">sig_extract()</a></code> is provided.</li>
<li>User now can directly use <code><a href="../reference/sig_extract.html">sig_extract()</a></code> and <code><a href="../reference/sig_auto_extract.html">sig_auto_extract()</a></code> instead of loading NMF package firstly.</li>
<li>Added benchmark results for different extraction approaches in README.</li>
<li>The threshold for <code>auto_reduce</code> in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> is modified from 0.99 to 0.95 and similarity update threshold updated from <code>&gt;0</code> to <code>&gt;=0.01</code>.</li>
<li>Removed <code>pConstant</code> option from <code><a href="../reference/sig_extract.html">sig_extract()</a></code> and <code><a href="../reference/sig_estimate.html">sig_estimate()</a></code>. Now a auto-check function is created for avoiding the error from NMF package due to no contribution of a component in all samples.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-1-2-0">NEW FUNCTIONS<a class="anchor" aria-label="anchor" href="#new-functions-1-2-0"></a></h3>
<ul><li>
<code><a href="../reference/bp.html">bp_show_survey2()</a></code> to plot a simplified version for signature number survey (<a href="https://github.com/ShixiangWang/sigminer/issues/330" class="external-link">#330</a>).</li>
<li>
<code><a href="../reference/read_xena_variants.html">read_xena_variants()</a></code> to read variant data from UCSC Xena as a <code>MAF</code> object for signature analysis.</li>
<li>
<code><a href="../reference/get_sig_rec_similarity.html">get_sig_rec_similarity()</a></code> for getting reconstructed profile similarity for <code>Signature</code> object (<a href="https://github.com/ShixiangWang/sigminer/issues/293" class="external-link">#293</a>).</li>
<li>Added functions start with <code>bp_</code> which are combined to provide a best practice for extracting signatures in cancer researches. See more details, run <code><a href="../reference/bp.html">?bp</a></code> in your R console.</li>
</ul></div>
<div class="section level3">
<h3 id="deprecated-1-2-0">DEPRECATED<a class="anchor" aria-label="anchor" href="#deprecated-1-2-0"></a></h3>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.0" id="sigminer-110">sigminer 1.1.0<small>2020-11-11</small><a class="anchor" aria-label="anchor" href="#sigminer-110"></a></h2>
<ul><li>Added data simulation.</li>
<li>Suppressed <code>future</code> warnings.</li>
<li>Fixed p value calculation in bootstrap analysis.</li>
<li>Fixed typo in <code><a href="../reference/show_cor.html">show_cor()</a></code>, thanks to <a href="https://github.com/Miachol" class="external-link">@Miachol</a>.</li>
<li>Added <code>y_tr</code> option in <code><a href="../reference/show_sig_profile.html">show_sig_profile()</a></code> to transform y axis values.</li>
<li>Optimized default behavior of <code><a href="../reference/read_copynumber.html">read_copynumber()</a></code>.
<ul><li>Support LOH records when user input minor allele copy number.</li>
<li>Set <code>complement = FALSE</code> as default.</li>
<li>Free dependencies between option <code>use_all</code> and <code>complement</code>.</li>
</ul></li>
<li>Added visualization support for genome rearrangement signatures (<a href="https://github.com/ShixiangWang/sigminer/issues/300" class="external-link">#300</a>).</li>
<li>Added four database for reference signatures from <a href="https://doi.org/10.1038/s43018-020-0027-5" class="external-link uri">https://doi.org/10.1038/s43018-020-0027-5</a> (<a href="https://github.com/ShixiangWang/sigminer/issues/299" class="external-link">#299</a>).</li>
<li>Added new measure ‘CV’ for <code><a href="../reference/show_sig_bootstrap.html">show_sig_bootstrap()</a></code> (<a href="https://github.com/ShixiangWang/sigminer/issues/298" class="external-link">#298</a>).</li>
<li>Added <code><a href="../reference/group_enrichment.html">group_enrichment()</a></code> and <code><a href="../reference/show_group_enrichment.html">show_group_enrichment()</a></code> (<a href="https://github.com/ShixiangWang/sigminer/issues/277" class="external-link">#277</a>).</li>
<li>Optimized signature profile visualization (<a href="https://github.com/ShixiangWang/sigminer/issues/295" class="external-link">#295</a>).</li>
<li>Updated <code><a href="../reference/sigminer-package.html">?sigminer</a></code> documentation.</li>
<li>Added <code>ms</code> strategy to select optimal solution by maximizing cosine similarity to reference signatures.</li>
<li>Added <code><a href="../reference/same_size_clustering.html">same_size_clustering()</a></code> for same size clustering.</li>
<li>Added <code><a href="../reference/show_cosmic.html">show_cosmic()</a></code> to support reading COSMIC signatures in web browser (<a href="https://github.com/ShixiangWang/sigminer/issues/288" class="external-link">#288</a>).</li>
<li>Changed argument <code>rel_threshold</code> behavior in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> and <code><a href="../reference/get_sig_exposure.html">get_sig_exposure()</a></code>. Made them more consistent and allowed un-assigned signature contribution (<a href="https://github.com/ShixiangWang/sigminer/issues/285" class="external-link">#285</a>).</li>
<li>Updated all COSMIC signatures to v3.1 and their aetiologies (<a href="https://github.com/ShixiangWang/sigminer/issues/287" class="external-link">#287</a>).</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.19" id="sigminer-1019">sigminer 1.0.19<small>2020-09-28</small><a class="anchor" aria-label="anchor" href="#sigminer-1019"></a></h2>
<ul><li>Added more specific reference signatures from SigProfiler, e.g. <code>SBS_mm9</code>.</li>
<li>Supported <code>data.frame</code> as input object for <code>sig</code> in <code><a href="../reference/get_sig_similarity.html">get_sig_similarity()</a></code> and <code><a href="../reference/sig_fit.html">sig_fit()</a></code>.</li>
<li>Modified <code>g_label</code> option in <code><a href="../reference/show_group_distribution.html">show_group_distribution()</a></code> to better control group names.</li>
<li>Added <code>test</code> option and variable checking in <code><a href="../reference/show_cor.html">show_cor()</a></code>.</li>
<li>Updated <code><a href="../reference/output_sig.html">output_sig()</a></code> to output signature exposure distribution (<a href="https://github.com/ShixiangWang/sigminer/issues/280" class="external-link">#280</a>).</li>
<li>Added <code><a href="../reference/show_cor.html">show_cor()</a></code> for general association analysis.</li>
<li>Added options in <code><a href="../reference/show_group_distribution.html">show_group_distribution()</a></code> to control segments.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.18" id="sigminer-1018">sigminer 1.0.18<a class="anchor" aria-label="anchor" href="#sigminer-1018"></a></h2>
<ul><li>Fixed bugs when outputing only 1 signatures.</li>
<li>Fixed label inverse bug in <code><a href="../reference/add_labels.html">add_labels()</a></code>, thanks to TaoTao for reporting.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.17" id="sigminer-1017">sigminer 1.0.17<a class="anchor" aria-label="anchor" href="#sigminer-1017"></a></h2>
<ul><li>Handled <code>,</code> seperated indices in show_cosmic_signatures.</li>
<li>Added option <code>set_order</code> in <code><a href="../reference/get_sig_similarity.html">get_sig_similarity()</a></code> (<a href="https://github.com/ShixiangWang/sigminer/issues/274" class="external-link">#274</a>).</li>
<li>Outputed more stats information in <code><a href="../reference/output_sig.html">output_sig()</a></code>.</li>
<li>Fixed default y axis title in <code><a href="../reference/show_sig_bootstrap.html">show_sig_bootstrap_error()</a></code>, now it is “Reconstruction error (L2 norm)”</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.16" id="sigminer-1016">sigminer 1.0.16<small>2020-09-12</small><a class="anchor" aria-label="anchor" href="#sigminer-1016"></a></h2>
<ul><li>Added <code>auto_reduce</code> option in <code>sig_fit*</code> functions to improve signature fitting.</li>
<li>Return cosine similarity for sample profile in <code><a href="../reference/sig_fit.html">sig_fit()</a></code>.</li>
<li>Set default strategy in <code><a href="../reference/sig_auto_extract.html">sig_auto_extract()</a></code> to ‘optimal’.</li>
<li>Supported search reference signature index in <code><a href="../reference/get_sig_cancer_type_index.html">get_sig_cancer_type_index()</a></code>.</li>
<li>Outputed legacy COSMIC similarity for SBS signatures.</li>
<li>Added new option in <code><a href="../reference/sigprofiler.html">sigprofiler_extract()</a></code> to reduce failure in when <code>refit</code> is enabled.</li>
<li>Outputed both relative and absolute signature exposure in <code><a href="../reference/output_sig.html">output_sig()</a></code>.</li>
<li>Updated background color in <code><a href="../reference/show_group_distribution.html">show_group_distribution()</a></code>.</li>
<li>Modified the default theme for signature profile in COSMIC style.</li>
<li>Updated the copy number classification method.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.15" id="sigminer-1015">sigminer 1.0.15<a class="anchor" aria-label="anchor" href="#sigminer-1015"></a></h2>
<ul><li>Handled null catalogue.</li>
<li>Supported ordering the signatures for results from SigProfiler.</li>
<li>Supported importing refit results from SigProfiler.</li>
<li>Set <code>optimize</code> option in <code><a href="../reference/sig_extract.html">sig_extract()</a></code> and <code><a href="../reference/sig_auto_extract.html">sig_auto_extract()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.14" id="sigminer-1014">sigminer 1.0.14<a class="anchor" aria-label="anchor" href="#sigminer-1014"></a></h2>
<ul><li>Supported signature index separated by <code>,</code> in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> and <code>sig_fit_bootstrap*</code> functions.</li>
<li>Added <code>output_*</code> functions from <a href="https://github.com/ShixiangWang/sigflow" class="external-link">sigflow</a>.</li>
<li>Enhanced DBS search and error handling in <code><a href="../reference/sig_tally.html">sig_tally()</a></code>.</li>
<li>Added option <code>highlight_genes</code> in <code><a href="../reference/show_cn_group_profile.html">show_cn_group_profile()</a></code> to show gene labels.</li>
<li>Added <code><a href="../reference/get_sig_cancer_type_index.html">get_sig_cancer_type_index()</a></code> to get reference signature index.</li>
<li>Added <code><a href="../reference/show_group_distribution.html">show_group_distribution()</a></code> to show group distribution.</li>
<li>Added options in <code><a href="../reference/show_cn_profile.html">show_cn_profile()</a></code> to show specified ranges and add copy number value labels.</li>
<li>Used package <code>nnls</code> instead of <code>pracma</code> for NNLS implementation in <code><a href="../reference/sig_fit.html">sig_fit()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.13" id="sigminer-1013">sigminer 1.0.13<small>2020-08-27</small><a class="anchor" aria-label="anchor" href="#sigminer-1013"></a></h2>
<ul><li>Supported <code>BSgenome.Hsapiens.1000genomes.hs37d5</code> in <code><a href="../reference/sig_tally.html">sig_tally()</a></code>.</li>
<li>Remove changing <code>MT</code> to <code>M</code> in mutation data.</li>
<li>Fixed bug in extract numeric signature names and signature orderings in <code><a href="../reference/show_sig_exposure.html">show_sig_exposure()</a></code>.</li>
<li>Added <code>letter_colors</code> as an unexported discrete palette.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.12" id="sigminer-1012">sigminer 1.0.12<a class="anchor" aria-label="anchor" href="#sigminer-1012"></a></h2>
<ul><li>Added <code><a href="../reference/transform_seg_table.html">transform_seg_table()</a></code>.</li>
<li>Added <code><a href="../reference/show_cn_group_profile.html">show_cn_group_profile()</a></code>.</li>
<li>Added <code><a href="../reference/show_cn_freq_circos.html">show_cn_freq_circos()</a></code>.</li>
<li>
<code>sig_orders</code> option in <code><a href="../reference/show_sig_profile.html">show_sig_profile()</a></code> function now can select and order signatures to plot.</li>
<li>Added <code><a href="../reference/show_sig_profile_loop.html">show_sig_profile_loop()</a></code> for better signature profile visualization.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.11" id="sigminer-1011">sigminer 1.0.11<a class="anchor" aria-label="anchor" href="#sigminer-1011"></a></h2>
<ul><li>Added option to control the SigProfilerExtractor to avoid issue in docker image build.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.10" id="sigminer-1010">sigminer 1.0.10<a class="anchor" aria-label="anchor" href="#sigminer-1010"></a></h2>
<ul><li>Some updates.</li>
<li>Compatible with SigProfiler 1.0.15</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.9" id="sigminer-109">sigminer 1.0.9<a class="anchor" aria-label="anchor" href="#sigminer-109"></a></h2>
<ul><li>Tried to speed up joining adjacent segments in <code><a href="../reference/read_copynumber.html">read_copynumber()</a></code>, got 200% improvement.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.8" id="sigminer-108">sigminer 1.0.8<a class="anchor" aria-label="anchor" href="#sigminer-108"></a></h2>
<ul><li>Tried to speed up joining adjacent segments in <code><a href="../reference/read_copynumber.html">read_copynumber()</a></code>, got 20% improvement.</li>
<li>Added <code><a href="../reference/cosine.html">cosine()</a></code> function.</li>
<li>Added and exported <code><a href="../reference/get_sig_db.html">get_sig_db()</a></code> to let users directly load signature database.</li>
<li>Added <code><a href="../reference/sigprofiler.html">sigprofiler_extract()</a></code> and <code><a href="../reference/sigprofiler.html">sigprofiler_import()</a></code> to call SigProfiler and import results.</li>
<li>Added <code><a href="../reference/read_vcf.html">read_vcf()</a></code> for simply reading VCF files.</li>
<li>Implemented DBS-1248.</li>
<li>Added <code><a href="../reference/show_sig_profile_heatmap.html">show_sig_profile_heatmap()</a></code>.</li>
<li>Supported mouse genome ‘mm10’ (<a href="https://github.com/ShixiangWang/sigminer/issues/241" class="external-link">#241</a>).</li>
<li>Added <code><a href="../reference/read_copynumber_seqz.html">read_copynumber_seqz()</a></code> to read sequenza result directory.</li>
<li>Speed up the annotation process in <code><a href="../reference/read_copynumber.html">read_copynumber()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.7" id="sigminer-107">sigminer 1.0.7<small>2020-06-17</small><a class="anchor" aria-label="anchor" href="#sigminer-107"></a></h2>
<ul><li>Fixed bug in OsCN feature calculation.</li>
<li>Removed useless options in <code><a href="../reference/read_maf.html">read_maf()</a></code>.</li>
<li>Modify method ‘LS’ in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> to ‘NNLS’ and implement it with <strong>pracma</strong> package (<a href="https://github.com/ShixiangWang/sigminer/issues/216" class="external-link">#216</a>).</li>
<li>Made <code>use_all</code> option in <code><a href="../reference/read_copynumber.html">read_copynumber()</a></code> working correctly.</li>
<li>Fixed potential problem raised by unordered copy number segments (<a href="https://github.com/ShixiangWang/sigminer/issues/217" class="external-link">#217</a>).</li>
<li>Fixed a typo, correct <code>MRSE</code> to <code>RMSE</code>.</li>
<li>Added feature in <code>show_sig_bootstrap_*()</code> for plotting aggregated values.</li>
<li>Fixed bug when use <code><a href="../reference/get_groups.html">get_groups()</a></code> for clustering.</li>
<li>Fixed bug about using reference components from NatGen 2018 paper.</li>
<li>Added option <code>highlight_size</code> for <code>show_sig_bootstrap_*()</code>.</li>
<li>Fixed bug about signature profile plotting for method ‘M’.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.6" id="sigminer-106">sigminer 1.0.6<small>2020-06-01</small><a class="anchor" aria-label="anchor" href="#sigminer-106"></a></h2>
<ul><li>Added “scatter” in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> function to better visualize a few samples.</li>
<li>Added “highlight” option.</li>
<li>
<code>lsei</code> package was removed from CRAN, here I reset default method to ‘QP’ and tried best to keep the LS usage in sigminer (<a href="https://github.com/ShixiangWang/sigminer/issues/189" class="external-link">#189</a>).</li>
<li>Made consistent copy number labels in <code><a href="../reference/show_sig_profile.html">show_sig_profile()</a></code> and added input checking for this function.</li>
<li>Fixed unconsistent bootstrap when use <code>furrr</code>, solution is from <a href="https://github.com/DavisVaughan/furrr/issues/107" class="external-link uri">https://github.com/DavisVaughan/furrr/issues/107</a>.</li>
<li>Properly handled null-count sample in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> for methods <code>QP</code> and <code>SA</code>.</li>
<li>Supported boxplot or violin in <code><a href="../reference/show_sig_fit.html">show_sig_fit()</a></code> and <code>show_sig_bootstrap_*</code> functions.</li>
<li>Added job mode for <code>sig_fit_bootstrap_batch</code> for more useful in practice.</li>
<li>Added <code><a href="../reference/show_groups.html">show_groups()</a></code> to show the signature contribution in each group from <code><a href="../reference/get_groups.html">get_groups()</a></code>.</li>
<li>Expanded clustering in <code><a href="../reference/get_groups.html">get_groups()</a></code> to result of <code><a href="../reference/sig_fit.html">sig_fit()</a></code>.</li>
<li>Properly handled null-count samples in <code><a href="../reference/sig_fit_bootstrap_batch.html">sig_fit_bootstrap_batch()</a></code>.</li>
<li>Added strand bias labeling for INDEL.</li>
<li>Added COSMIC TSB signatures.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.5" id="sigminer-105">sigminer 1.0.5<small>2020-05-14</small><a class="anchor" aria-label="anchor" href="#sigminer-105"></a></h2>
<ul><li>Exported APOBEC result when the mode is ‘ALL’ in <code><a href="../reference/sig_tally.html">sig_tally()</a></code>.</li>
<li>Added batch bootstrap analysis feature (<a href="https://github.com/ShixiangWang/sigminer/issues/158" class="external-link">#158</a>).</li>
<li>Supported all common signature plotting.</li>
<li>Added strand feature to signature profile.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.4" id="sigminer-104">sigminer 1.0.4<a class="anchor" aria-label="anchor" href="#sigminer-104"></a></h2>
<ul><li>Added profile plot for DBS and INDEL.</li>
<li>Fixed error for signature extraction in mode ‘DBS’ or ‘ID’.</li>
<li>Fixed method ‘M’ for CN tally cannot work when <code>cores &gt; 1</code> (<a href="https://github.com/ShixiangWang/sigminer/issues/161" class="external-link">#161</a>).</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.3" id="sigminer-103">sigminer 1.0.3<small>2020-04-30</small><a class="anchor" aria-label="anchor" href="#sigminer-103"></a></h2>
<ul><li>Added multiple methods for <code><a href="../reference/sig_fit.html">sig_fit()</a></code>.</li>
<li>Added feature <code><a href="../reference/sig_fit_bootstrap.html">sig_fit_bootstrap()</a></code> for bootstrap results.</li>
<li>Added multiple classification method for SBS signature.</li>
<li>Added strand bias enrichment analysis for SBS signature.</li>
<li>Moved multiple packages from field <code>Imports</code> to <code>Suggests</code>.</li>
<li>Added feature <code><a href="../reference/report_bootstrap_p_value.html">report_bootstrap_p_value()</a></code> to report p values.</li>
<li>Added common DBS and ID signature.</li>
<li>Updated citation.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.2" id="sigminer-102">sigminer 1.0.2<a class="anchor" aria-label="anchor" href="#sigminer-102"></a></h2>
<ul><li>Added merged transcript info for hg19 and hg38 build, this is availabe by <code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code>.</li>
<li>Added gene info for hg19 and hg38 build to extdata directory.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.1" id="sigminer-101">sigminer 1.0.1<a class="anchor" aria-label="anchor" href="#sigminer-101"></a></h2>
<ul><li>Removed <code>fuzzyjoin</code> package from dependency.</li>
<li>Moved <code>ggalluvial</code> package to field <code>suggsets</code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.0" id="sigminer-100">sigminer 1.0.0<small>2020-03-31</small><a class="anchor" aria-label="anchor" href="#sigminer-100"></a></h2>
<p>All users, this is a break-through version of <strong>sigminer</strong>, most of functions have been modified, more features are implemented. Please read the reference list to see the function groups and their functionalities.</p>
<p>Please read the vignette for usage.</p>
<p>I Hope it helps your research work and makes a new contribution to the scientific community.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://orcid.org/0000-0001-9855-7357" class="external-link">Shixiang Wang</a>, Ziyu Tao, Huimin Li, Tao Wu, <a href="https://orcid.org/0000-0002-7736-0077" class="external-link">Xue-Song Liu</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

