<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Changelog • sigminer</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Changelog" />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sigminer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.0.99</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/sigminer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ShixiangWang/sigminer/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href='https://github.com/ShixiangWang/sigminer/blob/master/NEWS.md'><code>NEWS.md</code></a></small>
    </div>

    <div id="sigminer-120" class="section level1">
<h1 class="page-header" data-toc-text="1.2.0">
<a href="#sigminer-120" class="anchor"></a>sigminer 1.2.0<small> Unreleased </small>
</h1>
<div id="bug-reports" class="section level2">
<h2 class="hasAnchor">
<a href="#bug-reports" class="anchor"></a>BUG REPORTS</h2>
</div>
<div id="enhancements" class="section level2">
<h2 class="hasAnchor">
<a href="#enhancements" class="anchor"></a>ENHANCEMENTS</h2>
<ul>
<li>A Python backend for <code><a href="../reference/sig_extract.html">sig_extract()</a></code> is provided.</li>
<li>User now can directly use <code><a href="../reference/sig_extract.html">sig_extract()</a></code> and <code><a href="../reference/sig_auto_extract.html">sig_auto_extract()</a></code> instead of loading NMF package firstly.</li>
<li>Added benchmark results for different extraction approaches in README.</li>
<li>The threshold for <code>auto_reduce</code> in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> is modified from 0.99 to 0.95 and similarity update threshold updated from <code>&gt;0</code> to <code>&gt;=0.01</code>.</li>
<li>Removed <code>pConstant</code> option from <code><a href="../reference/sig_extract.html">sig_extract()</a></code> and <code><a href="../reference/sig_estimate.html">sig_estimate()</a></code>. Now a auto-check function is created for avoiding the error from NMF package due to no contribution of a component in all samples.</li>
</ul>
</div>
<div id="new-functions" class="section level2">
<h2 class="hasAnchor">
<a href="#new-functions" class="anchor"></a>NEW FUNCTIONS</h2>
<ul>
<li>Added functions start with <code>bp_</code> which are combined to provide a best practice for extracting signatures in cancer researches. See more details, run <code><a href="../reference/bp.html">?bp</a></code> in your R console.</li>
</ul>
</div>
<div id="deprecated" class="section level2">
<h2 class="hasAnchor">
<a href="#deprecated" class="anchor"></a>DEPRECATED</h2>
</div>
</div>
    <div id="sigminer-110" class="section level1">
<h1 class="page-header" data-toc-text="1.1.0">
<a href="#sigminer-110" class="anchor"></a>sigminer 1.1.0<small> 2020-11-11 </small>
</h1>
<ul>
<li>Added data simulation.</li>
<li>Suppressed <code>future</code> warnings.</li>
<li>Fixed p value calculation in bootstrap analysis.</li>
<li>Fixed typo in <code><a href="../reference/show_cor.html">show_cor()</a></code>, thanks to <a href='https://github.com/Miachol'>@Miachol</a>.</li>
<li>Added <code>y_tr</code> option in <code><a href="../reference/show_sig_profile.html">show_sig_profile()</a></code> to transform y axis values.</li>
<li>Optimized default behavior of <code><a href="../reference/read_copynumber.html">read_copynumber()</a></code>.
<ul>
<li>Support LOH records when user input minor allele copy number.</li>
<li>Set <code>complement = FALSE</code> as default.</li>
<li>Free dependencies between option <code>use_all</code> and <code>complement</code>.</li>
</ul>
</li>
<li>Added visualization support for genome rearrangement signatures (<a href='https://github.com/ShixiangWang/sigminer/issues/300'>#300</a>).</li>
<li>Added four database for reference signatures from <a href="https://doi.org/10.1038/s43018-020-0027-5" class="uri">https://doi.org/10.1038/s43018-020-0027-5</a> (<a href='https://github.com/ShixiangWang/sigminer/issues/299'>#299</a>).</li>
<li>Added new measure ‘CV’ for <code><a href="../reference/show_sig_bootstrap.html">show_sig_bootstrap()</a></code> (<a href='https://github.com/ShixiangWang/sigminer/issues/298'>#298</a>).</li>
<li>Added <code><a href="../reference/group_enrichment.html">group_enrichment()</a></code> and <code><a href="../reference/show_group_enrichment.html">show_group_enrichment()</a></code> (<a href='https://github.com/ShixiangWang/sigminer/issues/277'>#277</a>).</li>
<li>Optimized signature profile visualization (<a href='https://github.com/ShixiangWang/sigminer/issues/295'>#295</a>).</li>
<li>Updated <code><a href="../reference/sigminer.html">?sigminer</a></code> documentation.</li>
<li>Added <code>ms</code> strategy to select optimal solution by maximizing cosine similarity to reference signatures.</li>
<li>Added <code><a href="../reference/same_size_clustering.html">same_size_clustering()</a></code> for same size clustering.</li>
<li>Added <code><a href="../reference/show_cosmic.html">show_cosmic()</a></code> to support reading COSMIC signatures in web browser (<a href='https://github.com/ShixiangWang/sigminer/issues/288'>#288</a>).</li>
<li>Changed argument <code>rel_threshold</code> behavior in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> and <code><a href="../reference/get_sig_exposure.html">get_sig_exposure()</a></code>. Made them more consistent and allowed un-assigned signature contribution (<a href='https://github.com/ShixiangWang/sigminer/issues/285'>#285</a>).</li>
<li>Updated all COSMIC signatures to v3.1 and their aetiologies (<a href='https://github.com/ShixiangWang/sigminer/issues/287'>#287</a>).</li>
</ul>
</div>
    <div id="sigminer-1019" class="section level1">
<h1 class="page-header" data-toc-text="1.0.19">
<a href="#sigminer-1019" class="anchor"></a>sigminer 1.0.19<small> 2020-09-28 </small>
</h1>
<ul>
<li>Added more specific reference signatures from SigProfiler, e.g. <code>SBS_mm9</code>.</li>
<li>Supported <code>data.frame</code> as input object for <code>sig</code> in <code><a href="../reference/get_sig_similarity.html">get_sig_similarity()</a></code> and <code><a href="../reference/sig_fit.html">sig_fit()</a></code>.</li>
<li>Modified <code>g_label</code> option in <code><a href="../reference/show_group_distribution.html">show_group_distribution()</a></code> to better control group names.</li>
<li>Added <code>test</code> option and variable checking in <code><a href="../reference/show_cor.html">show_cor()</a></code>.</li>
<li>Updated <code><a href="../reference/output_sig.html">output_sig()</a></code> to output signature exposure distribution (<a href="https://github.com/ShixiangWang/sigminer/issues/280"><a href='https://github.com/ShixiangWang/sigminer/issues/280'>#280</a></a>).</li>
<li>Added <code><a href="../reference/show_cor.html">show_cor()</a></code> for general association analysis.</li>
<li>Added options in <code><a href="../reference/show_group_distribution.html">show_group_distribution()</a></code> to control segments.</li>
</ul>
</div>
    <div id="sigminer-1018" class="section level1">
<h1 class="page-header" data-toc-text="1.0.18">
<a href="#sigminer-1018" class="anchor"></a>sigminer 1.0.18<small> Unreleased </small>
</h1>
<ul>
<li>Fixed bugs when outputing only 1 signatures.</li>
<li>Fixed label inverse bug in <code><a href="../reference/add_labels.html">add_labels()</a></code>, thanks to TaoTao for reporting.</li>
</ul>
</div>
    <div id="sigminer-1017" class="section level1">
<h1 class="page-header" data-toc-text="1.0.17">
<a href="#sigminer-1017" class="anchor"></a>sigminer 1.0.17<small> Unreleased </small>
</h1>
<ul>
<li>Handled <code>,</code> seperated indices in show_cosmic_signatures.</li>
<li>Added option <code>set_order</code> in <code><a href="../reference/get_sig_similarity.html">get_sig_similarity()</a></code> (<a href='https://github.com/ShixiangWang/sigminer/issues/274'>#274</a>).</li>
<li>Outputed more stats information in <code><a href="../reference/output_sig.html">output_sig()</a></code>.</li>
<li>Fixed default y axis title in <code><a href="../reference/show_sig_bootstrap.html">show_sig_bootstrap_error()</a></code>, now it is “Reconstruction error (L2 norm)”</li>
</ul>
</div>
    <div id="sigminer-1016" class="section level1">
<h1 class="page-header" data-toc-text="1.0.16">
<a href="#sigminer-1016" class="anchor"></a>sigminer 1.0.16<small> 2020-09-12 </small>
</h1>
<ul>
<li>Added <code>auto_reduce</code> option in <code>sig_fit*</code> functions to improve signature fitting.</li>
<li>Return cosine similarity for sample profile in <code><a href="../reference/sig_fit.html">sig_fit()</a></code>.</li>
<li>Set default strategy in <code><a href="../reference/sig_auto_extract.html">sig_auto_extract()</a></code> to ‘optimal’.</li>
<li>Supported search reference signature index in <code><a href="../reference/get_sig_cancer_type_index.html">get_sig_cancer_type_index()</a></code>.</li>
<li>Outputed legacy COSMIC similarity for SBS signatures.</li>
<li>Added new option in <code><a href="../reference/sigprofiler.html">sigprofiler_extract()</a></code> to reduce failure in when <code>refit</code> is enabled.</li>
<li>Outputed both relative and absolute signature exposure in <code><a href="../reference/output_sig.html">output_sig()</a></code>.</li>
<li>Updated background color in <code><a href="../reference/show_group_distribution.html">show_group_distribution()</a></code>.</li>
<li>Modified the default theme for signature profile in COSMIC style.</li>
<li>Updated the copy number classification method.</li>
</ul>
</div>
    <div id="sigminer-1015" class="section level1">
<h1 class="page-header" data-toc-text="1.0.15">
<a href="#sigminer-1015" class="anchor"></a>sigminer 1.0.15<small> Unreleased </small>
</h1>
<ul>
<li>Handled null catalogue.</li>
<li>Supported ordering the signatures for results from SigProfiler.</li>
<li>Supported importing refit results from SigProfiler.</li>
<li>Set <code>optimize</code> option in <code><a href="../reference/sig_extract.html">sig_extract()</a></code> and <code><a href="../reference/sig_auto_extract.html">sig_auto_extract()</a></code>.</li>
</ul>
</div>
    <div id="sigminer-1014" class="section level1">
<h1 class="page-header" data-toc-text="1.0.14">
<a href="#sigminer-1014" class="anchor"></a>sigminer 1.0.14<small> Unreleased </small>
</h1>
<ul>
<li>Supported signature index separated by <code>,</code> in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> and <code>sig_fit_bootstrap*</code> functions.</li>
<li>Added <code>output_*</code> functions from <a href="https://github.com/ShixiangWang/sigflow">sigflow</a>.</li>
<li>Enhanced DBS search and error handling in <code><a href="../reference/sig_tally.html">sig_tally()</a></code>.</li>
<li>Added option <code>highlight_genes</code> in <code><a href="../reference/show_cn_group_profile.html">show_cn_group_profile()</a></code> to show gene labels.</li>
<li>Added <code><a href="../reference/get_sig_cancer_type_index.html">get_sig_cancer_type_index()</a></code> to get reference signature index.</li>
<li>Added <code><a href="../reference/show_group_distribution.html">show_group_distribution()</a></code> to show group distribution.</li>
<li>Added options in <code><a href="../reference/show_cn_profile.html">show_cn_profile()</a></code> to show specified ranges and add copy number value labels.</li>
<li>Used package <code>nnls</code> instead of <code>pracma</code> for NNLS implementation in <code><a href="../reference/sig_fit.html">sig_fit()</a></code>.</li>
</ul>
</div>
    <div id="sigminer-1013" class="section level1">
<h1 class="page-header" data-toc-text="1.0.13">
<a href="#sigminer-1013" class="anchor"></a>sigminer 1.0.13<small> 2020-08-27 </small>
</h1>
<ul>
<li>Supported <code>BSgenome.Hsapiens.1000genomes.hs37d5</code> in <code><a href="../reference/sig_tally.html">sig_tally()</a></code>.</li>
<li>Remove changing <code>MT</code> to <code>M</code> in mutation data.</li>
<li>Fixed bug in extract numeric signature names and signature orderings in <code><a href="../reference/show_sig_exposure.html">show_sig_exposure()</a></code>.</li>
<li>Added <code>letter_colors</code> as an unexported discrete palette.</li>
</ul>
</div>
    <div id="sigminer-1012" class="section level1">
<h1 class="page-header" data-toc-text="1.0.12">
<a href="#sigminer-1012" class="anchor"></a>sigminer 1.0.12<small> Unreleased </small>
</h1>
<ul>
<li>Added <code><a href="../reference/transform_seg_table.html">transform_seg_table()</a></code>.</li>
<li>Added <code><a href="../reference/show_cn_group_profile.html">show_cn_group_profile()</a></code>.</li>
<li>Added <code><a href="../reference/show_cn_freq_circos.html">show_cn_freq_circos()</a></code>.</li>
<li>
<code>sig_orders</code> option in <code><a href="../reference/show_sig_profile.html">show_sig_profile()</a></code> function now can select and order signatures to plot.</li>
<li>Added <code><a href="../reference/show_sig_profile_loop.html">show_sig_profile_loop()</a></code> for better signature profile visualization.</li>
</ul>
</div>
    <div id="sigminer-1011" class="section level1">
<h1 class="page-header" data-toc-text="1.0.11">
<a href="#sigminer-1011" class="anchor"></a>sigminer 1.0.11<small> Unreleased </small>
</h1>
<ul>
<li>Added option to control the SigProfilerExtractor to avoid issue in docker image build.</li>
</ul>
</div>
    <div id="sigminer-1010" class="section level1">
<h1 class="page-header" data-toc-text="1.0.10">
<a href="#sigminer-1010" class="anchor"></a>sigminer 1.0.10<small> Unreleased </small>
</h1>
<ul>
<li>Some updates.</li>
<li>Compatible with SigProfiler 1.0.15</li>
</ul>
</div>
    <div id="sigminer-109" class="section level1">
<h1 class="page-header" data-toc-text="1.0.9">
<a href="#sigminer-109" class="anchor"></a>sigminer 1.0.9<small> Unreleased </small>
</h1>
<ul>
<li>Tried to speed up joining adjacent segments in <code><a href="../reference/read_copynumber.html">read_copynumber()</a></code>, got 200% improvement.</li>
</ul>
</div>
    <div id="sigminer-108" class="section level1">
<h1 class="page-header" data-toc-text="1.0.8">
<a href="#sigminer-108" class="anchor"></a>sigminer 1.0.8<small> Unreleased </small>
</h1>
<ul>
<li>Tried to speed up joining adjacent segments in <code><a href="../reference/read_copynumber.html">read_copynumber()</a></code>, got 20% improvement.</li>
<li>Added <code><a href="../reference/cosine.html">cosine()</a></code> function.</li>
<li>Added and exported <code><a href="../reference/get_sig_db.html">get_sig_db()</a></code> to let users directly load signature database.</li>
<li>Added <code><a href="../reference/sigprofiler.html">sigprofiler_extract()</a></code> and <code><a href="../reference/sigprofiler.html">sigprofiler_import()</a></code> to call SigProfiler and import results.</li>
<li>Added <code><a href="../reference/read_vcf.html">read_vcf()</a></code> for simply reading VCF files.</li>
<li>Implemented DBS-1248.</li>
<li>Added <code><a href="../reference/show_sig_profile_heatmap.html">show_sig_profile_heatmap()</a></code>.</li>
<li>Supported mouse genome ‘mm10’ (<a href="https://github.com/ShixiangWang/sigminer/issues/241"><a href='https://github.com/ShixiangWang/sigminer/issues/241'>#241</a></a>).</li>
<li>Added <code><a href="../reference/read_copynumber_seqz.html">read_copynumber_seqz()</a></code> to read sequenza result directory.</li>
<li>Speed up the annotation process in <code><a href="../reference/read_copynumber.html">read_copynumber()</a></code>.</li>
</ul>
</div>
    <div id="sigminer-107" class="section level1">
<h1 class="page-header" data-toc-text="1.0.7">
<a href="#sigminer-107" class="anchor"></a>sigminer 1.0.7<small> 2020-06-17 </small>
</h1>
<ul>
<li>Fixed bug in OsCN feature calculation.</li>
<li>Removed useless options in <code><a href="../reference/read_maf.html">read_maf()</a></code>.</li>
<li>Modify method ‘LS’ in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> to ‘NNLS’ and implement it with <strong>pracma</strong> package (<a href="https://github.com/ShixiangWang/sigminer/issues/216"><a href='https://github.com/ShixiangWang/sigminer/issues/216'>#216</a></a>).</li>
<li>Made <code>use_all</code> option in <code><a href="../reference/read_copynumber.html">read_copynumber()</a></code> working correctly.</li>
<li>Fixed potential problem raised by unordered copy number segments (<a href="https://github.com/ShixiangWang/sigminer/issues/217"><a href='https://github.com/ShixiangWang/sigminer/issues/217'>#217</a></a>).</li>
<li>Fixed a typo, correct <code>MRSE</code> to <code>RMSE</code>.</li>
<li>Added feature in <code>show_sig_bootstrap_*()</code> for plotting aggregated values.</li>
<li>Fixed bug when use <code><a href="../reference/get_groups.html">get_groups()</a></code> for clustering.</li>
<li>Fixed bug about using reference components from NatGen 2018 paper.</li>
<li>Added option <code>highlight_size</code> for <code>show_sig_bootstrap_*()</code>.</li>
<li>Fixed bug about signature profile plotting for method ‘M’.</li>
</ul>
</div>
    <div id="sigminer-106" class="section level1">
<h1 class="page-header" data-toc-text="1.0.6">
<a href="#sigminer-106" class="anchor"></a>sigminer 1.0.6<small> 2020-06-01 </small>
</h1>
<ul>
<li>Added “scatter” in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> function to better visualize a few samples.</li>
<li>Added “highlight” option.</li>
<li>
<code>lsei</code> package was removed from CRAN, here I reset default method to ‘QP’ and tried best to keep the LS usage in sigminer (<a href="https://github.com/ShixiangWang/sigminer/issues/189"><a href='https://github.com/ShixiangWang/sigminer/issues/189'>#189</a></a>).</li>
<li>Made consistent copy number labels in <code><a href="../reference/show_sig_profile.html">show_sig_profile()</a></code> and added input checking for this function.</li>
<li>Fixed unconsistent bootstrap when use <code>furrr</code>, solution is from <a href="https://github.com/DavisVaughan/furrr/issues/107" class="uri">https://github.com/DavisVaughan/furrr/issues/107</a>.</li>
<li>Properly handled null-count sample in <code><a href="../reference/sig_fit.html">sig_fit()</a></code> for methods <code>QP</code> and <code>SA</code>.</li>
<li>Supported boxplot or violin in <code><a href="../reference/show_sig_fit.html">show_sig_fit()</a></code> and <code>show_sig_bootstrap_*</code> functions.</li>
<li>Added job mode for <code>sig_fit_bootstrap_batch</code> for more useful in practice.</li>
<li>Added <code><a href="../reference/show_groups.html">show_groups()</a></code> to show the signature contribution in each group from <code><a href="../reference/get_groups.html">get_groups()</a></code>.</li>
<li>Expanded clustering in <code><a href="../reference/get_groups.html">get_groups()</a></code> to result of <code><a href="../reference/sig_fit.html">sig_fit()</a></code>.</li>
<li>Properly handled null-count samples in <code><a href="../reference/sig_fit_bootstrap_batch.html">sig_fit_bootstrap_batch()</a></code>.</li>
<li>Added strand bias labeling for INDEL.</li>
<li>Added COSMIC TSB signatures.</li>
</ul>
</div>
    <div id="sigminer-105" class="section level1">
<h1 class="page-header" data-toc-text="1.0.5">
<a href="#sigminer-105" class="anchor"></a>sigminer 1.0.5<small> 2020-05-14 </small>
</h1>
<ul>
<li>Exported APOBEC result when the mode is ‘ALL’ in <code><a href="../reference/sig_tally.html">sig_tally()</a></code>.</li>
<li>Added batch bootstrap analysis feature (<a href='https://github.com/ShixiangWang/sigminer/issues/158'>#158</a>).</li>
<li>Supported all common signature plotting.</li>
<li>Added strand feature to signature profile.</li>
</ul>
</div>
    <div id="sigminer-104" class="section level1">
<h1 class="page-header" data-toc-text="1.0.4">
<a href="#sigminer-104" class="anchor"></a>sigminer 1.0.4<small> Unreleased </small>
</h1>
<ul>
<li>Added profile plot for DBS and INDEL.</li>
<li>Fixed error for signature extraction in mode ‘DBS’ or ‘ID’.</li>
<li>Fixed method ‘M’ for CN tally cannot work when <code>cores &gt; 1</code> (<a href='https://github.com/ShixiangWang/sigminer/issues/161'>#161</a>).</li>
</ul>
</div>
    <div id="sigminer-103" class="section level1">
<h1 class="page-header" data-toc-text="1.0.3">
<a href="#sigminer-103" class="anchor"></a>sigminer 1.0.3<small> 2020-04-30 </small>
</h1>
<ul>
<li>Added multiple methods for <code><a href="../reference/sig_fit.html">sig_fit()</a></code>.</li>
<li>Added feature <code><a href="../reference/sig_fit_bootstrap.html">sig_fit_bootstrap()</a></code> for bootstrap results.</li>
<li>Added multiple classification method for SBS signature.</li>
<li>Added strand bias enrichment analysis for SBS signature.</li>
<li>Moved multiple packages from field <code>Imports</code> to <code>Suggests</code>.</li>
<li>Added feature <code><a href="../reference/report_bootstrap_p_value.html">report_bootstrap_p_value()</a></code> to report p values.</li>
<li>Added common DBS and ID signature.</li>
<li>Updated citation.</li>
</ul>
</div>
    <div id="sigminer-102" class="section level1">
<h1 class="page-header" data-toc-text="1.0.2">
<a href="#sigminer-102" class="anchor"></a>sigminer 1.0.2<small> Unreleased </small>
</h1>
<ul>
<li>Added merged transcript info for hg19 and hg38 build, this is availabe by <code><a href="https://rdrr.io/r/utils/data.html">data()</a></code>.</li>
<li>Added gene info for hg19 and hg38 build to extdata directory.</li>
</ul>
</div>
    <div id="sigminer-101" class="section level1">
<h1 class="page-header" data-toc-text="1.0.1">
<a href="#sigminer-101" class="anchor"></a>sigminer 1.0.1<small> Unreleased </small>
</h1>
<ul>
<li>Removed <code>fuzzyjoin</code> package from dependency.</li>
<li>Moved <code>ggalluvial</code> package to field <code>suggsets</code>.</li>
</ul>
</div>
    <div id="sigminer-100" class="section level1">
<h1 class="page-header" data-toc-text="1.0.0">
<a href="#sigminer-100" class="anchor"></a>sigminer 1.0.0<small> 2020-03-31 </small>
</h1>
<p>All users, this is a break-through version of <strong>sigminer</strong>, most of functions have been modified, more features are implemented. Please read the reference list to see the function groups and their functionalities.</p>
<p>Please read the vignette for usage.</p>
<p>I Hope it helps your research work and makes a new contribution to the scientific community.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://orcid.org/0000-0001-9855-7357'>Shixiang Wang</a>, <a href='https://orcid.org/0000-0002-7736-0077'>Xue-Song Liu</a>, Tao Wu, Ziyu Tao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


