<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Read Absolute Copy Number Profile — read_copynumber • sigminer</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Read Absolute Copy Number Profile — read_copynumber"><meta property="og:description" content="Read absolute copy number profile for preparing CNV signature
analysis. See detail part of sig_tally() to see how to handle sex to get correct
summary."><meta property="og:image" content="https://shixiangwang.github.io/sigminer/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sigminer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/sigminer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cnsignature.html">Analyze Copy Number Signatures with sigminer</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ShixiangWang/sigminer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read Absolute Copy Number Profile</h1>
    <small class="dont-index">Source: <a href="https://github.com/ShixiangWang/sigminer/blob/master/R/read_copynumber.R" class="external-link"><code>R/read_copynumber.R</code></a></small>
    <div class="hidden name"><code>read_copynumber.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Read <strong>absolute</strong> copy number profile for preparing CNV signature
analysis. See detail part of <code><a href="sig_tally.html">sig_tally()</a></code> to see how to handle sex to get correct
summary.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">read_copynumber</span><span class="op">(</span></span>
<span>  <span class="va">input</span>,</span>
<span>  pattern <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ignore_case <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  seg_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="st">"Start.bp"</span>, <span class="st">"End.bp"</span>, <span class="st">"modal_cn"</span><span class="op">)</span>,</span>
<span>  samp_col <span class="op">=</span> <span class="st">"sample"</span>,</span>
<span>  add_loh <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  loh_min_len <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  loh_min_frac <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  join_adj_seg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  skip_annotation <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  use_all <span class="op">=</span> <span class="va">add_loh</span>,</span>
<span>  min_segnum <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  max_copynumber <span class="op">=</span> <span class="fl">20L</span>,</span>
<span>  genome_build <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hg19"</span>, <span class="st">"hg38"</span>, <span class="st">"T2T"</span>, <span class="st">"mm10"</span>, <span class="st">"mm9"</span>, <span class="st">"ce11"</span><span class="op">)</span>,</span>
<span>  genome_measure <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"called"</span>, <span class="st">"wg"</span><span class="op">)</span>,</span>
<span>  complement <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-input">input<a class="anchor" aria-label="anchor" href="#arg-input"></a></dt>
<dd><p>a <code>data.frame</code> or a file or a directory contains copy number profile.</p></dd>


<dt id="arg-pattern">pattern<a class="anchor" aria-label="anchor" href="#arg-pattern"></a></dt>
<dd><p>an optional regular expression used to select part of files if
<code>input</code> is a directory, more detail please see <code><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files()</a></code> function.</p></dd>


<dt id="arg-ignore-case">ignore_case<a class="anchor" aria-label="anchor" href="#arg-ignore-case"></a></dt>
<dd><p>logical. Should pattern-matching be case-insensitive?</p></dd>


<dt id="arg-seg-cols">seg_cols<a class="anchor" aria-label="anchor" href="#arg-seg-cols"></a></dt>
<dd><p>four strings used to specify chromosome, start position,
end position and copy number value in <code>input</code>, respectively.
Default use names from ABSOLUTE calling result.</p></dd>


<dt id="arg-samp-col">samp_col<a class="anchor" aria-label="anchor" href="#arg-samp-col"></a></dt>
<dd><p>a character used to specify the sample column name. If <code>input</code>
is a directory and cannot find <code>samp_col</code>, sample names will use file names
(set this parameter to <code>NULL</code> is recommended in this case).</p></dd>


<dt id="arg-add-loh">add_loh<a class="anchor" aria-label="anchor" href="#arg-add-loh"></a></dt>
<dd><p>if <code>TRUE</code>, add LOH labels to segments. <strong>NOTE</strong> a column
'minor_cn' must exist to indicate minor allele copy number value.
Sex chromosome will not be labeled.</p></dd>


<dt id="arg-loh-min-len">loh_min_len<a class="anchor" aria-label="anchor" href="#arg-loh-min-len"></a></dt>
<dd><p>The length cut-off for labeling a segment as 'LOH'.
Default is <code>10Kb</code>.</p></dd>


<dt id="arg-loh-min-frac">loh_min_frac<a class="anchor" aria-label="anchor" href="#arg-loh-min-frac"></a></dt>
<dd><p>When <code>join_adj_seg</code> set to <code>TRUE</code>, only the length fraction
of LOH region is larger than this value will be labeled as 'LOH'.
Default is 30%.</p></dd>


<dt id="arg-join-adj-seg">join_adj_seg<a class="anchor" aria-label="anchor" href="#arg-join-adj-seg"></a></dt>
<dd><p>if <code>TRUE</code> (default), join adjacent segments with
same copy number value. This is helpful for precisely count the number of breakpoint.
When set <code>use_all=TRUE</code>, the mean function will be applied to extra numeric columns
and unique string columns will be pasted by comma for joined records.</p></dd>


<dt id="arg-skip-annotation">skip_annotation<a class="anchor" aria-label="anchor" href="#arg-skip-annotation"></a></dt>
<dd><p>if <code>TRUE</code>, skip annotation step, it may affect some analysis
and visualization functionality, but speed up reading data.</p></dd>


<dt id="arg-use-all">use_all<a class="anchor" aria-label="anchor" href="#arg-use-all"></a></dt>
<dd><p>default is <code>FALSE</code>. If <code>True</code>, use all columns from raw input.</p></dd>


<dt id="arg-min-segnum">min_segnum<a class="anchor" aria-label="anchor" href="#arg-min-segnum"></a></dt>
<dd><p>minimal number of copy number segments within a sample.</p></dd>


<dt id="arg-max-copynumber">max_copynumber<a class="anchor" aria-label="anchor" href="#arg-max-copynumber"></a></dt>
<dd><p>bigger copy number within a sample will be reset to this value.</p></dd>


<dt id="arg-genome-build">genome_build<a class="anchor" aria-label="anchor" href="#arg-genome-build"></a></dt>
<dd><p>genome build version, should be 'hg19', 'hg38', 'mm9' or 'mm10'.</p></dd>


<dt id="arg-genome-measure">genome_measure<a class="anchor" aria-label="anchor" href="#arg-genome-measure"></a></dt>
<dd><p>default is 'called', can be 'wg' or 'called'.
Set 'called' will use called segments size to compute total size for CNA burden calculation,
this option is useful for WES and target sequencing.
Set 'wg' will use autosome size from genome build, this option is useful for WGS, SNP etc..</p></dd>


<dt id="arg-complement">complement<a class="anchor" aria-label="anchor" href="#arg-complement"></a></dt>
<dd><p>if <code>TRUE</code>, complement chromosome (except 'Y') does not show in input data
with normal copy 2.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>other parameters pass to <code><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">data.table::fread()</a></code></p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>a <a href="CopyNumber-class.html">CopyNumber</a> object.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="read_maf.html">read_maf</a> for reading mutation data to <a href="MAF-class.html">MAF</a> object.</p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Load toy dataset of absolute copynumber profile</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"toy_segTab.RData"</span>,</span></span>
<span class="r-in"><span>  package <span class="op">=</span> <span class="st">"sigminer"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="va">cn</span> <span class="op">&lt;-</span> <span class="fu">read_copynumber</span><span class="op">(</span><span class="va">segTabs</span>,</span></span>
<span class="r-in"><span>  seg_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chromosome"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"segVal"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  genome_build <span class="op">=</span> <span class="st">"hg19"</span>, complement <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.844451]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.846315]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg19.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.84789]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.852313]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.853929]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.855543]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.85721]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.858824]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.862154]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.866184]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.867784]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.869392]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 467</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.870953]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.872851]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.874424]: </span><span style="color: #0000BB; font-weight: bold;">Joining adjacent segments with same copy number value. Be patient...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.957191]: </span><span style="color: #00BB00; font-weight: bold;">400 segments left after joining.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.959095]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.960698]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.974678]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.976339]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.987448]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.989064]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.990848]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.992558]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.994158]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:05.995739]: </span><span style="color: #0000BB; font-weight: bold;">0.151 secs elapsed.</span></span>
<span class="r-in"><span><span class="va">cn</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An object of class CopyNumber </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> =============================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           sample n_of_seg n_of_cnv n_of_amp n_of_del n_of_vchr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           &lt;char&gt;    &lt;int&gt;    &lt;int&gt;    &lt;int&gt;    &lt;int&gt;     &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: TCGA-DF-A2KN-01A-11D-A17U-01       33        6        5        1         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: TCGA-19-2621-01B-01D-0911-01       33        8        5        3         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: TCGA-B6-A0X5-01A-21D-A107-01       28        8        4        4         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: TCGA-A8-A07S-01A-11D-A036-01       38       11        2        9         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: TCGA-26-6174-01A-21D-1842-01       43       13        8        5         8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: TCGA-CV-7432-01A-11D-2128-01       40       16        7        9         9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: TCGA-06-0644-01A-02D-0310-01       46       19        5       14         8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: TCGA-A5-A0G2-01A-11D-A042-01       39       21        5       16        10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: TCGA-99-7458-01A-11D-2035-01       48       26       10       16        13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10: TCGA-05-4417-01A-22D-1854-01       52       37       33        4        17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cna_burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:      0.000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:      0.099</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:      0.087</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:      0.112</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:      0.119</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:      0.198</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:      0.165</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:      0.393</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:      0.318</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:      0.654</span>
<span class="r-in"><span><span class="va">cn_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cn</span>, <span class="va">sample</span> <span class="op">==</span> <span class="st">"TCGA-DF-A2KN-01A-11D-A17U-01"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Add LOH</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">segTabs</span><span class="op">$</span><span class="va">minor_cn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">segTabs</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">cn</span> <span class="op">&lt;-</span> <span class="fu">read_copynumber</span><span class="op">(</span><span class="va">segTabs</span>,</span></span>
<span class="r-in"><span>  seg_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chromosome"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"segVal"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  genome_measure <span class="op">=</span> <span class="st">"wg"</span>, complement <span class="op">=</span> <span class="cn">TRUE</span>, add_loh <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.010834]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.012478]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg19.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.014039]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: wg.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.01565]: </span><span style="color: #0000BB; font-weight: bold;">When add_loh is TRUE, use_all is forced to TRUE.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB; font-weight: bold;">Please drop columns you don't want to keep before reading.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.019599]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.021248]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.028954]: </span><span style="color: #00BB00; font-weight: bold;">A data frame as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.031399]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.033215]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.036538]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.050504]: </span><span style="color: #00BB00; font-weight: bold;">Value 2 (normal copy) filled to uncalled chromosomes.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.054363]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.055968]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.057596]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 477</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.059175]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.061082]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.06264]: </span><span style="color: #0000BB; font-weight: bold;">Adding LOH labels...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.064766]: </span><span style="color: #0000BB; font-weight: bold;">Joining adjacent segments with same copy number value. Be patient...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.188114]: </span><span style="color: #00BB00; font-weight: bold;">410 segments left after joining.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.190287]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.191985]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.206178]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.207818]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.221452]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.223144]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.224955]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.226552]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.228141]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.229746]: </span><span style="color: #0000BB; font-weight: bold;">0.219 secs elapsed.</span></span>
<span class="r-in"><span><span class="co"># Use tally method "S" (Steele et al.)</span></span></span>
<span class="r-in"><span><span class="va">tally_s</span> <span class="op">&lt;-</span> <span class="fu"><a href="sig_tally.html">sig_tally</a></span><span class="op">(</span><span class="va">cn</span>, method <span class="op">=</span> <span class="st">"S"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.231884]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.235482]: </span><span style="color: #0000BB; font-weight: bold;">When you use method 'S', please make sure you have set 'join_adj_seg' to FALSE and 'add_loh' to TRUE in 'read_copynumber() in the previous step!</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.456921]: </span><span style="color: #00BB00; font-weight: bold;">Matrix generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.459281]: </span><span style="color: #0000BB; font-weight: bold;">0.227 secs elapsed.</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">tab_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"metastatic_tumor.segtab.txt"</span>,</span></span>
<span class="r-in"><span>  package <span class="op">=</span> <span class="st">"sigminer"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">cn2</span> <span class="op">&lt;-</span> <span class="fu">read_copynumber</span><span class="op">(</span><span class="va">tab_file</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.462441]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.464114]: </span><span style="color: #0000BB; font-weight: bold;">Genome build  : hg19.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.465684]: </span><span style="color: #0000BB; font-weight: bold;">Genome measure: called.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.470135]: </span><span style="color: #00BB00; font-weight: bold;">Chromosome size database for build obtained.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.471721]: </span><span style="color: #0000BB; font-weight: bold;">Reading input.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.473284]: </span><span style="color: #00BB00; font-weight: bold;">A file as input detected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.475599]: </span><span style="color: #00BB00; font-weight: bold;">Column names checked.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.477364]: </span><span style="color: #00BB00; font-weight: bold;">Column order set.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.480472]: </span><span style="color: #00BB00; font-weight: bold;">Chromosomes unified.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.484475]: </span><span style="color: #00BB00; font-weight: bold;">Data imported.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.486061]: </span><span style="color: #0000BB; font-weight: bold;">Segments info:</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.487596]: </span><span style="color: #0000BB; font-weight: bold;">    Keep - 201</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.489137]: </span><span style="color: #0000BB; font-weight: bold;">    Drop - 0</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.490946]: </span><span style="color: #00BB00; font-weight: bold;">Segments sorted.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.492526]: </span><span style="color: #0000BB; font-weight: bold;">Joining adjacent segments with same copy number value. Be patient...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.509899]: </span><span style="color: #00BB00; font-weight: bold;">180 segments left after joining.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.511667]: </span><span style="color: #00BB00; font-weight: bold;">Segmental table cleaned.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.513233]: </span><span style="color: #0000BB; font-weight: bold;">Annotating.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.526558]: </span><span style="color: #00BB00; font-weight: bold;">Annotation done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.528202]: </span><span style="color: #0000BB; font-weight: bold;">Summarizing per sample.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.538819]: </span><span style="color: #00BB00; font-weight: bold;">Summarized.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.540427]: </span><span style="color: #0000BB; font-weight: bold;">Generating CopyNumber object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.542213]: </span><span style="color: #00BB00; font-weight: bold;">Generated.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.543761]: </span><span style="color: #0000BB; font-weight: bold;">Validating object.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.545309]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-11-30 14:25:06.546874]: </span><span style="color: #0000BB; font-weight: bold;">0.084 secs elapsed.</span></span>
<span class="r-in"><span><span class="va">cn2</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An object of class CopyNumber </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> =============================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              sample n_of_seg n_of_cnv n_of_amp n_of_del n_of_vchr cna_burden</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              &lt;char&gt;    &lt;int&gt;    &lt;int&gt;    &lt;int&gt;    &lt;int&gt;     &lt;int&gt;      &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1: metastatic_tumor      180      110       92       18        21      0.361</span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://orcid.org/0000-0001-9855-7357" class="external-link">Shixiang Wang</a>, Ziyu Tao, Huimin Li, Tao Wu, <a href="https://orcid.org/0000-0002-7736-0077" class="external-link">Xue-Song Liu</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer></div>






  </body></html>

