<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Obtain Bootstrap Distribution of Signature Exposures of a Certain Tumor Sample — sig_fit_bootstrap • sigminer</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Obtain Bootstrap Distribution of Signature Exposures of a Certain Tumor Sample — sig_fit_bootstrap"><meta property="og:description" content="This can be used to obtain the confidence of signature exposures or search
the suboptimal decomposition solution."><meta property="og:image" content="https://shixiangwang.github.io/sigminer/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sigminer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/sigminer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cnsignature.html">Analyze Copy Number Signatures with sigminer</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ShixiangWang/sigminer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Obtain Bootstrap Distribution of Signature Exposures of a Certain Tumor Sample</h1>
    <small class="dont-index">Source: <a href="https://github.com/ShixiangWang/sigminer/blob/master/R/sig_fit_bootstrap.R" class="external-link"><code>R/sig_fit_bootstrap.R</code></a></small>
    <div class="hidden name"><code>sig_fit_bootstrap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This can be used to obtain the confidence of signature exposures or search
the suboptimal decomposition solution.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">sig_fit_bootstrap</span><span class="op">(</span></span>
<span>  <span class="va">catalog</span>,</span>
<span>  <span class="va">sig</span>,</span>
<span>  n <span class="op">=</span> <span class="fl">100L</span>,</span>
<span>  sig_index <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sig_db <span class="op">=</span> <span class="st">"legacy"</span>,</span>
<span>  db_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">""</span>, <span class="st">"human-exome"</span>, <span class="st">"human-genome"</span><span class="op">)</span>,</span>
<span>  show_index <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"QP"</span>, <span class="st">"NNLS"</span>, <span class="st">"SA"</span><span class="op">)</span>,</span>
<span>  auto_reduce <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  SA_not_bootstrap <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"absolute"</span>, <span class="st">"relative"</span><span class="op">)</span>,</span>
<span>  rel_threshold <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  mode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SBS"</span>, <span class="st">"DBS"</span>, <span class="st">"ID"</span>, <span class="st">"copynumber"</span><span class="op">)</span>,</span>
<span>  find_suboptimal <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  suboptimal_ref_error <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  suboptimal_factor <span class="op">=</span> <span class="fl">1.05</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-catalog">catalog<a class="anchor" aria-label="anchor" href="#arg-catalog"></a></dt>
<dd><p>a named numeric vector or a numeric matrix with dimension Nx1.
N is the number of component, 1 is the sample.</p></dd>


<dt id="arg-sig">sig<a class="anchor" aria-label="anchor" href="#arg-sig"></a></dt>
<dd><p>a <code>Signature</code> object obtained either from <a href="sig_extract.html">sig_extract</a> or <a href="sig_auto_extract.html">sig_auto_extract</a>,
or just a raw signature matrix/<code>data.frame</code> with row representing components (motifs) and
column representing signatures.</p></dd>


<dt id="arg-n">n<a class="anchor" aria-label="anchor" href="#arg-n"></a></dt>
<dd><p>the number of bootstrap replicates.</p></dd>


<dt id="arg-sig-index">sig_index<a class="anchor" aria-label="anchor" href="#arg-sig-index"></a></dt>
<dd><p>a vector for signature index. "ALL" for all signatures.</p></dd>


<dt id="arg-sig-db">sig_db<a class="anchor" aria-label="anchor" href="#arg-sig-db"></a></dt>
<dd><p>default 'legacy', it can be 'legacy' (for <a href="https://cancer.sanger.ac.uk/cosmic/signatures_v2.tt/" class="external-link">COSMIC v2 'SBS'</a>),
'SBS', 'DBS', 'ID' and 'TSB' (for <a href="https://cancer.sanger.ac.uk/cosmic/signatures/" class="external-link">COSMIV v3.1 signatures</a>)
for small scale mutations.
For more specific details, it can also be 'SBS_hg19', 'SBS_hg38',
'SBS_mm9', 'SBS_mm10', 'DBS_hg19', 'DBS_hg38', 'DBS_mm9', 'DBS_mm10' to use
COSMIC v3 reference signatures from Alexandrov, Ludmil B., et al. (2020) (reference #1).
In addition, it can be one of "SBS_Nik_lab_Organ", "RS_Nik_lab_Organ",
"SBS_Nik_lab", "RS_Nik_lab" to refer reference signatures from
Degasperi, Andrea, et al. (2020) (reference #2);
"RS_BRCA560", "RS_USARC" to reference signatures from BRCA560 and USARC cohorts;
"CNS_USARC" (40 categories), "CNS_TCGA" (48 categories) to reference copy number signatures from USARC cohort and TCGA;
"CNS_TCGA176" (176 categories) and "CNS_PCAWG176" (176 categories) to reference copy number signatures from PCAWG and TCGA separately.
<strong>UPDATE</strong>, the latest version of reference version can be automatically
downloaded and loaded from <a href="https://cancer.sanger.ac.uk/signatures/downloads/" class="external-link">https://cancer.sanger.ac.uk/signatures/downloads/</a>
when a option with <code>latest_</code> prefix is specified (e.g. "latest_SBS_GRCh37").
<strong>Note</strong>: the signature profile for different genome builds are basically same.
And specific database (e.g. 'SBS_mm10') contains less signatures than all COSMIC
signatures (because some signatures are not detected from Alexandrov, Ludmil B., et al. (2020)).
For all available options, check the parameter setting.</p></dd>


<dt id="arg-db-type">db_type<a class="anchor" aria-label="anchor" href="#arg-db-type"></a></dt>
<dd><p>only used when <code>sig_db</code> is enabled.
"" for keeping default, "human-exome" for transforming to exome frequency of component,
and "human-genome" for transforming to whole genome frequency of component.
Currently only works for 'SBS'.</p></dd>


<dt id="arg-show-index">show_index<a class="anchor" aria-label="anchor" href="#arg-show-index"></a></dt>
<dd><p>if <code>TRUE</code>, show valid indices.</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>method to solve the minimazation problem.
'NNLS' for non-negative least square; 'QP' for quadratic programming; 'SA' for simulated annealing.</p></dd>


<dt id="arg-auto-reduce">auto_reduce<a class="anchor" aria-label="anchor" href="#arg-auto-reduce"></a></dt>
<dd><p>if <code>TRUE</code>, try reducing the input reference signatures to increase
the cosine similarity of reconstructed profile to observed profile.</p></dd>


<dt id="arg-sa-not-bootstrap">SA_not_bootstrap<a class="anchor" aria-label="anchor" href="#arg-sa-not-bootstrap"></a></dt>
<dd><p>if <code>TRUE</code>, directly run 'SA' multiple times with original input instead of
bootstrap samples.</p></dd>


<dt id="arg-type">type<a class="anchor" aria-label="anchor" href="#arg-type"></a></dt>
<dd><p>'absolute' for signature exposure and 'relative' for signature relative exposure.</p></dd>


<dt id="arg-rel-threshold">rel_threshold<a class="anchor" aria-label="anchor" href="#arg-rel-threshold"></a></dt>
<dd><p>numeric vector, a signature with relative exposure
lower than (equal is included, i.e. <code>&lt;=</code>) this value will be set to 0
(both absolute exposure and relative exposure).
In this case, sum of signature contribution may not equal to 1.</p></dd>


<dt id="arg-mode">mode<a class="anchor" aria-label="anchor" href="#arg-mode"></a></dt>
<dd><p>signature type for plotting, now supports 'copynumber', 'SBS',
'DBS', 'ID' and 'RS' (genome rearrangement signature).</p></dd>


<dt id="arg-find-suboptimal">find_suboptimal<a class="anchor" aria-label="anchor" href="#arg-find-suboptimal"></a></dt>
<dd><p>logical, if <code>TRUE</code>, find suboptimal decomposition with
slightly higher error than the optimal solution by method 'SA'. This is useful
to explore hidden dependencies between signatures. More see reference.</p></dd>


<dt id="arg-suboptimal-ref-error">suboptimal_ref_error<a class="anchor" aria-label="anchor" href="#arg-suboptimal-ref-error"></a></dt>
<dd><p>baseline error used for finding suboptimal solution.
if it is <code>NULL</code>, then use 'SA' method to obtain the optimal error.</p></dd>


<dt id="arg-suboptimal-factor">suboptimal_factor<a class="anchor" aria-label="anchor" href="#arg-suboptimal-factor"></a></dt>
<dd><p>suboptimal factor to get suboptimal error, default is <code>1.05</code>,
i.e., suboptimal error is <code>1.05</code> times baseline error.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>control parameters passing to argument <code>control</code> in <code>GenSA</code> function when use method 'SA'.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>a <code>list</code></p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Huang X, Wojtowicz D, Przytycka TM. Detecting presence of mutational signatures in cancer with confidence. Bioinformatics. 2018;34(2):330–337. doi:10.1093/bioinformatics/btx604</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="report_bootstrap_p_value.html">report_bootstrap_p_value</a>, <a href="sig_fit.html">sig_fit</a>, <a href="sig_fit_bootstrap_batch.html">sig_fit_bootstrap_batch</a></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># This function is designed for processing</span></span></span>
<span class="r-in"><span><span class="co"># one sample, thus is not very useful in practice</span></span></span>
<span class="r-in"><span><span class="co"># please check `sig_fit_bootstrap_batch`</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># For general purpose -------------------</span></span></span>
<span class="r-in"><span><span class="va">W</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">6</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">W</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sig1"</span>, <span class="st">"sig2"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">W</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">W</span>, <span class="fl">2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">H</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">5</span>, <span class="fl">3</span>, <span class="fl">6</span>, <span class="fl">1</span>, <span class="fl">9</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">H</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"samp"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">V</span> <span class="op">&lt;-</span> <span class="va">W</span> <span class="op"><a href="https://rdrr.io/r/base/matmult.html" class="external-link">%*%</a></span> <span class="va">H</span></span></span>
<span class="r-in"><span><span class="va">V</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         samp1 samp2    samp3 samp4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1,] 1.666667   2.1 2.566667   0.7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2,] 2.333333   3.0 3.333333   1.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3,] 3.000000   3.9 4.100000   1.3</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"quadprog"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>  <span class="va">H_bootstrap</span> <span class="op">&lt;-</span> <span class="fu">sig_fit_bootstrap</span><span class="op">(</span><span class="va">V</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">W</span>, n <span class="op">=</span> <span class="fl">10</span>, type <span class="op">=</span> <span class="st">"absolute"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="co">## Typically, you have to run many times to get close to the answer</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">H_bootstrap</span><span class="op">$</span><span class="va">expo</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">H</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>  <span class="co">## Return P values</span></span></span>
<span class="r-in"><span>  <span class="co">## In practice, run times &gt;= 100</span></span></span>
<span class="r-in"><span>  <span class="co">## is recommended</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="report_bootstrap_p_value.html">report_bootstrap_p_value</a></span><span class="op">(</span><span class="va">H_bootstrap</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="co">## For multiple samples</span></span></span>
<span class="r-in"><span>  <span class="co">## Input a list</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="report_bootstrap_p_value.html">report_bootstrap_p_value</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>samp1 <span class="op">=</span> <span class="va">H_bootstrap</span>, samp2 <span class="op">=</span> <span class="va">H_bootstrap</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>  <span class="co">#   ## Find suboptimal decomposition</span></span></span>
<span class="r-in"><span>  <span class="co">#   H_suboptimal &lt;- sig_fit_bootstrap(V[, 1], W,</span></span></span>
<span class="r-in"><span>  <span class="co">#     n = 10,</span></span></span>
<span class="r-in"><span>  <span class="co">#     type = "absolute",</span></span></span>
<span class="r-in"><span>  <span class="co">#     method = "SA",</span></span></span>
<span class="r-in"><span>  <span class="co">#     find_suboptimal = TRUE</span></span></span>
<span class="r-in"><span>  <span class="co">#   )</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.418279]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.420193]: </span><span style="color: #0000BB; font-weight: bold;">Checking catalog.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.42184]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.423443]: </span><span style="color: #0000BB; font-weight: bold;">About to start bootstrap.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Bootstrapping 10 times.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Total 10 times, starting no.1.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Total 10 times, starting no.2.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Total 10 times, starting no.3.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Total 10 times, starting no.4.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Total 10 times, starting no.5.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Total 10 times, starting no.6.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Total 10 times, starting no.7.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Total 10 times, starting no.8.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Total 10 times, starting no.9.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Total 10 times, starting no.10.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Bootstrap done.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.732534]: </span><span style="color: #00BB00; font-weight: bold;">Signature exposures collected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.734272]: </span><span style="color: #00BB00; font-weight: bold;">Errors and similarity collected.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.735873]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.737516]: </span><span style="color: #0000BB; font-weight: bold;">0.319 secs elapsed.</span></span>
<span class="r-plt img"><img src="sig_fit_bootstrap-1.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.74233]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.744137]: </span><span style="color: #00BB00; font-weight: bold;">Single sample mode enabled.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.746801]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.748627]: </span><span style="color: #0000BB; font-weight: bold;">0.006 secs elapsed.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.750358]: </span><span style="color: #0000BB; font-weight: bold;">Started.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.752023]: </span><span style="color: #00BB00; font-weight: bold;">Batch mode enabled.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.755299]: </span><span style="color: #00BB00; font-weight: bold;">Done.</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> <span style="font-weight: bold;">[2025-08-22 03:15:08.757012]: </span><span style="color: #0000BB; font-weight: bold;">0.007 secs elapsed.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $samp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      threshold_0.01 threshold_0.05 threshold_0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sig1    0.018043105    0.027706812   0.047025270</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sig2    0.001655784    0.002473604   0.004135223</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $samp2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      threshold_0.01 threshold_0.05 threshold_0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sig1    0.018043105    0.027706812   0.047025270</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sig2    0.001655784    0.002473604   0.004135223</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://orcid.org/0000-0001-9855-7357" class="external-link">Shixiang Wang</a>, Ziyu Tao, Huimin Li, Tao Wu, <a href="https://orcid.org/0000-0002-7736-0077" class="external-link">Xue-Song Liu</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

