<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Extract, Analyze and Visualize Mutational Signatures for
    Genomic Variations • sigminer</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Extract, Analyze and Visualize Mutational Signatures for
    Genomic Variations">
<meta property="og:description" content="Genomic alterations including single nucleotide
    substitution, copy number alteration, etc. are the major force for
    cancer initialization and development. Due to the specificity of
    molecular lesions caused by genomic alterations, we can generate
    characteristic alteration spectra, called signature (Wang, Shixiang,
    et al.  (2020) &lt;DOI:10.1101/2020.04.27.20082404&gt; &amp; Alexandrov, Ludmil
    B., et al. (2020) &lt;DOI:10.1038/s41586-020-1943-3&gt; &amp; Macintyre, Geoff,
    et al. (2018) &lt;DOI:10.1038/s41588-018-0179-8&gt;).  This package helps
    users to extract, analyze and visualize signatures from genomic
    alteration records, thus providing new insight into cancer study.">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">sigminer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/sigminer.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ShixiangWang/sigminer/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="sigminer-mutational-signature-analysis-and-visualization-in-r-" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#sigminer-mutational-signature-analysis-and-visualization-in-r-" class="anchor"></a>Sigminer: Mutational Signature Analysis and Visualization in R <img src="reference/figures/logo.png" alt="logo" align="right" height="140" width="120">
</h1></div>

<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>The cancer genome is shaped by various mutational processes over its lifetime, stemming from exogenous and cell-intrinsic DNA damage, and error-prone DNA replication, leaving behind characteristic mutational spectra, termed <strong>mutational signatures</strong>. This package, <strong>sigminer</strong>, helps users to extract, analyze and visualize signatures from genome alteration records, thus providing new insight into cancer study.</p>
<p>For pipeline tool, please see its co-evolutionary CLI <a href="https://github.com/ShixiangWang/sigflow">sigflow</a>.</p>
<p><strong>SBS signatures</strong>:</p>
<p><img src="reference/figures/README-unnamed-chunk-1-1.png" width="100%"></p>
<p><strong>Copy number signatures</strong>:</p>
<p><img src="reference/figures/README-unnamed-chunk-2-1.png" width="100%"></p>
<p><strong>DBS signatures</strong>:</p>
<p><img src="reference/figures/README-unnamed-chunk-3-1.png" width="100%"></p>
<p><strong>INDEL (i.e. ID) signatures</strong>:</p>
<p><img src="reference/figures/README-unnamed-chunk-4-1.png" width="100%"></p>
<p><strong>Genome rearrangement signatures</strong>:</p>
<p><img src="reference/figures/README-unnamed-chunk-5-1.png" width="100%"></p>
<blockquote>
<p><code><a href="reference/sig_tally.html">sig_tally()</a></code> for RS is not supported.</p>
</blockquote>
<div id="features" class="section level3">
<h3 class="hasAnchor">
<a href="#features" class="anchor"></a>Features</h3>
<ul>
<li>supports a standard <em>de novo</em> pipeline for identification of <strong>5</strong> types of signatures: copy number, SBS, DBS, INDEL and RS (genome rearrangement signature).</li>
<li>supports quantify exposure for one sample based on <em>known signatures</em>.</li>
<li>supports two methods for calling copy number signatures: one is from Macintyre et al. 2018 and the other is created by our group.</li>
<li>supports association and group analysis and visualization for signatures.</li>
<li>supports two types of signature exposures: relative exposure (relative contribution of signatures in each sample) and absolute exposure (estimated variation records of signatures in each sample).</li>
<li>supports basic summary and visualization for profile of mutation (powered by <strong>maftools</strong>) and copy number.</li>
<li>supports parallel computation by R packages <strong>foreach</strong>, <strong>future</strong> and <strong>NMF</strong>.</li>
<li>efficient code powered by R packages <strong>data.table</strong> and <strong>tidyverse</strong>.</li>
<li>elegant plots powered by R packages <strong>ggplot2</strong>, <strong>ggpubr</strong>, <strong>cowplot</strong> and <strong>patchwork</strong>.</li>
<li>well tested by R package <strong>testthat</strong> and documented by R package <strong>roxygen2</strong>, <strong>roxytest</strong>, <strong>pkgdown</strong>, and etc. for both reliable and reproducible research.</li>
</ul>
</div>
<div id="key-interfaces-and-their-performances" class="section level3">
<h3 class="hasAnchor">
<a href="#key-interfaces-and-their-performances" class="anchor"></a>Key Interfaces and Their Performances</h3>
<p><strong>Sigminer</strong> provides many approaches to extract mutational signatures. To test their performances, I use 4 mutation catalog datasets (each mutation catalog dataset is composed of 30 samples, 10 COSMIC v2 (SBS) signatures are randomly assigned to each sample with random signature exposure) from reference #6. The following table shows how many signatures can be recovered and the corresponding average cosine similarity to COSMIC reference signatures for each approach with settings.</p>
<table class="table">
<colgroup>
<col width="3%">
<col width="5%">
<col width="13%">
<col width="7%">
<col width="2%">
<col width="4%">
<col width="2%">
<col width="2%">
<col width="4%">
<col width="2%">
<col width="3%">
<col width="4%">
<col width="40%">
</colgroup>
<thead><tr class="header">
<th align="left">Approach</th>
<th align="left">Selection Way</th>
<th align="left">Setting</th>
<th align="left">Caller</th>
<th align="left">Recommend</th>
<th align="left">Driver</th>
<th align="left">Set1</th>
<th align="left">Set2</th>
<th align="left">Set3</th>
<th align="left">Set4</th>
<th align="left">Success /Mean</th>
<th align="left">Run time</th>
<th>Note</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Standard NMF</td>
<td align="left">Manual</td>
<td align="left">Default. 50 runs (estimation) + 100 runs (extraction)</td>
<td align="left">
<code>sig_estimate</code>, <code>sig_extract</code>
</td>
<td align="left">YES ⭐⭐⭐</td>
<td align="left">R</td>
<td align="left">10 (0.884)</td>
<td align="left">10 (0.944)</td>
<td align="left">9 or 10 (0.998)</td>
<td align="left">10 (0.994)</td>
<td align="left">~90%/0.955</td>
<td align="left">~1min (8 cores)</td>
<td>This is a basic method, suitable for good mutation data with enough mutations.</td>
</tr>
<tr class="even">
<td align="left">SigProfiler</td>
<td align="left"><strong>Manual/Automatic</strong></td>
<td align="left">Default. 100 runs</td>
<td align="left"><code>sigprofiler_extract</code></td>
<td align="left">YES ⭐⭐⭐⭐</td>
<td align="left">Python/Anaconda</td>
<td align="left">10 (0.961)</td>
<td align="left">10 (0.999)</td>
<td align="left">10 (0.990)</td>
<td align="left">10 (0.997)</td>
<td align="left">100%/0.987</td>
<td align="left">~1h (8 cores)</td>
<td>A golden standard like approach in this field, but longer run time, and the requirement for Python environment and extra large packages reduce its popularity here.</td>
</tr>
<tr class="odd">
<td align="left">Best Practice</td>
<td align="left"><strong>Manual/Automatic</strong></td>
<td align="left">Use bootstrapped catalog (1000 runs)</td>
<td align="left"><code>bp_extract_signatures</code></td>
<td align="left">YES ⭐⭐⭐⭐⭐</td>
<td align="left">R</td>
<td align="left">10 (0.973)</td>
<td align="left">10 (0.990)</td>
<td align="left">10 (0.992)</td>
<td align="left">10 (0.971)</td>
<td align="left">100%/0.981</td>
<td align="left">~10min (8 cores)</td>
<td>My R implementation for methods from reference #5 and #6. Should be the best option here.</td>
</tr>
<tr class="even">
<td align="left">Best Practice</td>
<td align="left"><strong>Manual/Automatic</strong></td>
<td align="left">Use original catalog (1000 runs)</td>
<td align="left"><code>bp_extract_signatures</code></td>
<td align="left">NO <span class="emoji" data-emoji="star">⭐</span>
</td>
<td align="left">R</td>
<td align="left">10 (0.987)</td>
<td align="left">9 (0.985)</td>
<td align="left">10 (0.997)</td>
<td align="left">9 (0.987)</td>
<td align="left">50%/0.989</td>
<td align="left">~10min (8 cores)</td>
<td>This is created to compare with the approach with bootstrapped catalogs above and the standard NMF way.</td>
</tr>
<tr class="odd">
<td align="left">Bayesian NMF</td>
<td align="left"><strong>Automatic</strong></td>
<td align="left">L1KL+optimal (20 runs)</td>
<td align="left"><code>sig_auto_extract</code></td>
<td align="left">YES ⭐⭐⭐</td>
<td align="left">R</td>
<td align="left">10 (0.994)</td>
<td align="left">9 (0.997)</td>
<td align="left">9 (0.998)</td>
<td align="left">9 (0.999)</td>
<td align="left">25%/0.997</td>
<td align="left">~10min (8 cores)</td>
<td>The Bayesian NMF approach auto reduce the signature number to a proper value from a initial signature number, here is 20.</td>
</tr>
<tr class="even">
<td align="left">Bayesian NMF</td>
<td align="left"><strong>Automatic</strong></td>
<td align="left">L1KL+stable (20 runs)</td>
<td align="left"><code>sig_auto_extract</code></td>
<td align="left">YES ⭐⭐⭐⭐</td>
<td align="left">R</td>
<td align="left">10 (0.994)</td>
<td align="left">9 (0.997)</td>
<td align="left">10 (0.988)</td>
<td align="left">9 (0.999)</td>
<td align="left">50%/0.995</td>
<td align="left">~10min (8 cores)</td>
<td>See above.</td>
</tr>
<tr class="odd">
<td align="left">Bayesian NMF</td>
<td align="left"><strong>Automatic</strong></td>
<td align="left">L2KL+optimal (20 runs)</td>
<td align="left"><code>sig_auto_extract</code></td>
<td align="left">NO <span class="emoji" data-emoji="star">⭐</span>
</td>
<td align="left">R</td>
<td align="left">12 (0.990)</td>
<td align="left">13 (0.988)</td>
<td align="left">12 (0.902)</td>
<td align="left">12 (0.994)</td>
<td align="left">0%/0.969</td>
<td align="left">~10min (8 cores)</td>
<td>See above.</td>
</tr>
<tr class="even">
<td align="left">Bayesian NMF</td>
<td align="left"><strong>Automatic</strong></td>
<td align="left">L2KL+stable (20 runs)</td>
<td align="left"><code>sig_auto_extract</code></td>
<td align="left">NO <span class="emoji" data-emoji="star">⭐</span>
</td>
<td align="left">R</td>
<td align="left">12 (0.990)</td>
<td align="left">12 (0.988)</td>
<td align="left">12 (0.902)</td>
<td align="left">12 (0.994)</td>
<td align="left">0%/0.969</td>
<td align="left">~10min (8 cores)</td>
<td>See above.</td>
</tr>
<tr class="odd">
<td align="left">Bayesian NMF</td>
<td align="left"><strong>Automatic</strong></td>
<td align="left">L1WL2H+optimal (20 runs)</td>
<td align="left"><code>sig_auto_extract</code></td>
<td align="left">YES ⭐⭐⭐</td>
<td align="left">R</td>
<td align="left">9 (0.989)</td>
<td align="left">9 (0.999)</td>
<td align="left">9 (0.996)</td>
<td align="left">9 (1.000)</td>
<td align="left">0%/0.996</td>
<td align="left">~10min (8 cores)</td>
<td>See above.</td>
</tr>
<tr class="even">
<td align="left">Bayesian NMF</td>
<td align="left"><strong>Automatic</strong></td>
<td align="left">L1WL2H+stable (20 runs)</td>
<td align="left"><code>sig_auto_extract</code></td>
<td align="left">YES ⭐⭐⭐⭐</td>
<td align="left">R</td>
<td align="left">9 (0.989)</td>
<td align="left">9 (0.999)</td>
<td align="left">9 (0.996)</td>
<td align="left">9 (1.000)</td>
<td align="left">0%/0.996</td>
<td align="left">~10min (8 cores)</td>
<td>See above.</td>
</tr>
</tbody>
</table>
<blockquote>
<p>NOTE: although Bayesian NMF approach with L1KL or L1WL2H prior cannot recover all 10 signatures here, but it is close to the true answer from initial signature number 20 in a automatic way, and the result signatures are highly similar to reference signatures. This also reminds us that we could not use this method to find signatures with small contributions in tumors.</p>
</blockquote>
</div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install the stable release of <strong>sigminer</strong> from CRAN with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"sigminer"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co"># Or</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"sigminer"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>You can install the development version of <strong>sigminer</strong> from Github with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"ShixiangWang/sigminer"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co"># For Chinese users, run </span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_git.html">install_git</a></span><span class="op">(</span><span class="st">"https://gitee.com/ShixiangWang/sigminer"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>You can also install <strong>sigminer</strong> from conda <code>bioconda</code> channel with</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Please note version number of the bioconda release</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># You can install an individual environment firstly with</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># conda create -n sigminer</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># conda activate sigminer</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install -c bioconda -c conda-forge r-sigminer</span></code></pre></div>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<p>A complete documentation of <strong>sigminer</strong> can be read online at <a href="https://shixiangwang.github.io/sigminer-doc/" class="uri">https://shixiangwang.github.io/sigminer-doc/</a> (For Chinese users, you can also read it at <a href="https://shixiangwang.gitee.io/sigminer-doc/" class="uri">https://shixiangwang.gitee.io/sigminer-doc/</a>). All functions are well organized and documented at <a href="https://shixiangwang.github.io/sigminer/reference/index.html" class="uri">https://shixiangwang.github.io/sigminer/reference/index.html</a> (For Chinese users, you can also read it at <a href="https://shixiangwang.gitee.io/sigminer/reference/index.html" class="uri">https://shixiangwang.gitee.io/sigminer/reference/index.html</a>). For usage of a specific function <code>fun</code>, run <code>?fun</code> in your R console to see its documentation.</p>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<p>If you use <strong>sigminer</strong> in academic field, please cite one of the following papers.</p>
<hr>
<ul>
<li><p><strong><em>Wang, Shixiang, et al. “Copy number signature analyses in prostate cancer reveal distinct etiologies and clinical outcomes” medRxiv (2020)</em></strong> <a href="https://www.medrxiv.org/content/10.1101/2020.04.27.20082404v1" class="uri">https://www.medrxiv.org/content/10.1101/2020.04.27.20082404v1</a></p></li>
<li><p><strong><em>Shixiang Wang, Ziyu Tao, Tao Wu, Xue-Song Liu, Sigflow: An Automated And Comprehensive Pipeline For Cancer Genome Mutational Signature Analysis, Bioinformatics, btaa895</em></strong>, <a href="https://doi.org/10.1093/bioinformatics/btaa895" class="uri">https://doi.org/10.1093/bioinformatics/btaa895</a></p></li>
</ul>
<hr>
</div>
<div id="download-stats" class="section level2">
<h2 class="hasAnchor">
<a href="#download-stats" class="anchor"></a>Download Stats</h2>
<p><img src="reference/figures/README-unnamed-chunk-8-1.png" width="100%"></p>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<p>Please properly cite the following references when you are using any corresponding features. The references are also listed in the function documentation. Very thanks to the works, <strong>sigminer</strong> cannot be created without the giants.</p>
<ol>
<li>Mayakonda, Anand, et al. “Maftools: efficient and comprehensive analysis of somatic variants in cancer.” Genome research 28.11 (2018): 1747-1756.</li>
<li>Gaujoux, Renaud, and Cathal Seoighe. “A Flexible R Package for Nonnegative Matrix Factorization.”" BMC Bioinformatics 11, no. 1 (December 2010).</li>
<li>H. Wickham. ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York, 2016.</li>
<li>Kim, Jaegil, et al. “Somatic ERCC2 mutations are associated with a distinct genomic signature in urothelial tumors.” Nature genetics 48.6 (2016): 600.</li>
<li>Alexandrov, Ludmil B., et al. “Deciphering signatures of mutational processes operative in human cancer.” Cell reports 3.1 (2013): 246-259.</li>
<li>Degasperi, Andrea, et al. “A practical framework and online tool for mutational signature analyses show intertissue variation and driver dependencies.” Nature cancer 1.2 (2020): 249-263.</li>
<li>Alexandrov, Ludmil B., et al. “The repertoire of mutational signatures in human cancer.” Nature 578.7793 (2020): 94-101.</li>
<li>Macintyre, Geoff, et al. “Copy number signatures and mutational processes in ovarian carcinoma.” Nature genetics 50.9 (2018): 1262.</li>
<li>Tan, Vincent YF, and Cédric Févotte. “Automatic relevance determination in nonnegative matrix factorization with the/spl beta/-divergence.” IEEE Transactions on Pattern Analysis and Machine Intelligence 35.7 (2012): 1592-1605.</li>
<li>Bergstrom EN, Huang MN, Mahto U, Barnes M, Stratton MR, Rozen SG, Alexandrov LB: SigProfilerMatrixGenerator: a tool for visualizing and exploring patterns of small mutational events. BMC Genomics 2019, 20:685 <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6041-2" class="uri">https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6041-2</a>
</li>
</ol>
</div>
<div id="license" class="section level2">
<h2 class="hasAnchor">
<a href="#license" class="anchor"></a>LICENSE</h2>
<p>The software is made available for non commercial research purposes only under the <a href="https://github.com/ShixiangWang/sigminer/blob/master/LICENSE.md">MIT</a>. However, notwithstanding any provision of the MIT License, the software currently may not be used for commercial purposes without explicit written permission after contacting Shixiang Wang <a href="mailto:wangshx@shanghaitech.edu.cn" class="email">wangshx@shanghaitech.edu.cn</a> or Xue-Song Liu <a href="mailto:liuxs@shanghaitech.edu.cn" class="email">liuxs@shanghaitech.edu.cn</a>.</p>
<p>MIT © 2019-Present Shixiang Wang, Xue-Song Liu</p>
<p>MIT © 2018 Geoffrey Macintyre</p>
<p>MIT © 2018 Anand Mayakonda</p>
<hr>
<p><a href="https://github.com/XSLiuLab"><strong>Cancer Biology Group</strong></a> <strong>@ShanghaiTech</strong></p>
<p><strong>Research group led by Xue-Song Liu in ShanghaiTech University</strong></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=sigminer">https://​cloud.r-project.org/​package=sigminer</a>
</li>
<li>Browse source code at <br><a href="https://github.com/ShixiangWang/sigminer/">https://​github.com/​ShixiangWang/​sigminer/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/ShixiangWang/sigminer/issues">https://​github.com/​ShixiangWang/​sigminer/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing sigminer</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://orcid.org/0000-0001-9855-7357">Shixiang Wang</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-9855-7357" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Ziyu Tao <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-3272-1227" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Huimin Li <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-1683-9057" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Tao Wu <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-8999-9628" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>
<a href="https://orcid.org/0000-0002-7736-0077">Xue-Song Liu</a> <br><small class="roles"> Author, contributor </small> <a href="https://orcid.org/0000-0002-7736-0077" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://cran.r-project.org/package=sigminer"><img src="https://www.r-pkg.org/badges/version/sigminer" alt="CRAN status"></a></li>
<li><a href="https://lifecycle.r-lib.org/articles/stages.html"><img src="https://img.shields.io/badge/lifecycle-stable-blue.svg" alt="lifecycle"></a></li>
<li><a href="https://codecov.io/github/ShixiangWang/sigminer?branch=master"><img src="https://codecov.io/gh/ShixiangWang/sigminer/branch/master/graph/badge.svg" alt="Coverage status"></a></li>
<li><a href="https://cran.r-project.org/package=sigminer"><img src="https://cranlogs.r-pkg.org/badges/grand-total/sigminer?color=orange"></a></li>
<li><a href="https://github.com/ShixiangWang/sigminer/issues?q=is%3Aissue+is%3Aclosed"><img src="https://img.shields.io/github/issues-closed/ShixiangWang/sigminer.svg" alt="Closed issues"></a></li>
<li><a href="https://github.com/ShixiangWang/sigminer"><img src="https://tokei.rs/b1/github/ShixiangWang/sigminer?category=code" alt="Lines Of Code"></a></li>
<li><a href="https://hits.dwyl.com/ShixiangWang/sigminer/"><img src="https://hits.dwyl.com/ShixiangWang/sigminer.svg" alt="HitCount"></a></li>
<li><a href="https://anaconda.org/bioconda/r-sigminer/"><img src="https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat-square" alt="install with bioconda"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://orcid.org/0000-0001-9855-7357">Shixiang Wang</a>, Ziyu Tao, Huimin Li, Tao Wu, <a href="https://orcid.org/0000-0002-7736-0077">Xue-Song Liu</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
