<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ sigminer</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sigminer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/sigminer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ShixiangWang/sigminer/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-data-input-classes-and-operation" class="hasAnchor"><a href="#section-data-input-classes-and-operation" class="anchor"></a>Data input, classes and operation</h2>
          <p class="section-desc"><p>Read genomic variation data as an R object and more.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="read_copynumber.html">read_copynumber()</a></code> </p>
        </td>
        <td><p>Read Absolute Copy Number Profile</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_copynumber_seqz.html">read_copynumber_seqz()</a></code> </p>
        </td>
        <td><p>Read Absolute Copy Number Profile from Sequenza Result Directory</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_maf.html">read_maf()</a></code> </p>
        </td>
        <td><p>Read MAF Files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_sv_as_rs.html">read_sv_as_rs()</a></code> </p>
        </td>
        <td><p>Read Structural Variation Data as RS object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_vcf.html">read_vcf()</a></code> </p>
        </td>
        <td><p>Read VCF Files as MAF Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_xena_variants.html">read_xena_variants()</a></code> </p>
        </td>
        <td><p>Read UCSC Xena Variant Format Data as MAF Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CopyNumber-class.html">CopyNumber-class</a></code> </p>
        </td>
        <td><p>Class CopyNumber</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MAF-class.html">MAF-class</a></code> </p>
        </td>
        <td><p>Class MAF</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="subset.CopyNumber.html">subset(<i>&lt;CopyNumber&gt;</i>)</a></code> </p>
        </td>
        <td><p>Subsetting CopyNumber object</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-core-pipeline-de-novo-signature-identification" class="hasAnchor"><a href="#section-core-pipeline-de-novo-signature-identification" class="anchor"></a>Core pipeline 1 - de novo signature identification</h2>
          <p class="section-desc"><p>Identify signatures from genomic variation profile. After reading data, the following functions construct the standard pipeline for most users.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="sig_tally.html">sig_tally()</a></code> </p>
        </td>
        <td><p>Tally a Genomic Alteration Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="handle_hyper_mutation.html">handle_hyper_mutation()</a></code> </p>
        </td>
        <td><p>Handle Hypermutant Samples</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sig_estimate.html">sig_estimate()</a></code> <code><a href="sig_estimate.html">show_sig_number_survey()</a></code> <code><a href="sig_estimate.html">show_sig_number_survey2()</a></code> </p>
        </td>
        <td><p>Estimate Signature Number</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sig_extract.html">sig_extract()</a></code> </p>
        </td>
        <td><p>Extract Signatures through NMF</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sig_auto_extract.html">sig_auto_extract()</a></code> </p>
        </td>
        <td><p>Extract Signatures through the Automatic Relevance Determination Technique</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sigprofiler.html">sigprofiler_extract()</a></code> <code><a href="sigprofiler.html">sigprofiler_import()</a></code> </p>
        </td>
        <td><p>Extract Signatures with SigProfiler</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bp.html">bp_extract_signatures()</a></code> <code><a href="bp.html">bp_extract_signatures_iter()</a></code> <code><a href="bp.html">bp_cluster_iter_list()</a></code> <code><a href="bp.html">bp_get_clustered_sigs()</a></code> <code><a href="bp.html">bp_get_sig_obj()</a></code> <code><a href="bp.html">bp_get_stats()</a></code> <code><a href="bp.html">bp_get_rank_score()</a></code> <code><a href="bp.html">bp_show_survey2()</a></code> <code><a href="bp.html">bp_show_survey()</a></code> <code><a href="bp.html">bp_attribute_activity()</a></code> </p>
        </td>
        <td><p>A Best Practice for Signature Extraction and Exposure (Activity) Attribution</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-core-pipeline-signature-exposure-fitting-and-optimization" class="hasAnchor"><a href="#section-core-pipeline-signature-exposure-fitting-and-optimization" class="anchor"></a>Core pipeline 2 - signature exposure fitting and optimization</h2>
          <p class="section-desc"><p>Quantify exposure in at least one sample with reference signatures. It can also be used to optimize signature exposure from pipeline 1 and analyze exposure instability.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="sig_tally.html">sig_tally()</a></code> </p>
        </td>
        <td><p>Tally a Genomic Alteration Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="handle_hyper_mutation.html">handle_hyper_mutation()</a></code> </p>
        </td>
        <td><p>Handle Hypermutant Samples</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_sig_cancer_type_index.html">get_sig_cancer_type_index()</a></code> </p>
        </td>
        <td><p>Obtain Signature Index for Cancer Types</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sig_fit.html">sig_fit()</a></code> </p>
        </td>
        <td><p>Fit Signature Exposures with Linear Combination Decomposition</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sig_fit_bootstrap.html">sig_fit_bootstrap()</a></code> </p>
        </td>
        <td><p>Obtain Bootstrap Distribution of Signature Exposures of a Certain Tumor Sample</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sig_fit_bootstrap_batch.html">sig_fit_bootstrap_batch()</a></code> </p>
        </td>
        <td><p>Exposure Instability Analysis of Signature Exposures with Bootstrapping</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="report_bootstrap_p_value.html">report_bootstrap_p_value()</a></code> </p>
        </td>
        <td><p>Report P Values from bootstrap Results</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-signature-visualization" class="hasAnchor"><a href="#section-signature-visualization" class="anchor"></a>Signature visualization</h2>
          <p class="section-desc"><p>Show signature profile and exposure.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="show_catalogue.html">show_catalogue()</a></code> </p>
        </td>
        <td><p>Show Alteration Catalogue Profile</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_sig_profile.html">show_sig_profile()</a></code> </p>
        </td>
        <td><p>Show Signature Profile</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_sig_profile_loop.html">show_sig_profile_loop()</a></code> </p>
        </td>
        <td><p>Show Signature Profile with Loop Way</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_sig_profile_heatmap.html">show_sig_profile_heatmap()</a></code> </p>
        </td>
        <td><p>Show Signature Profile with Heatmap</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_sig_exposure.html">show_sig_exposure()</a></code> </p>
        </td>
        <td><p>Plot Signature Exposure</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_cosmic.html">show_cosmic()</a></code> </p>
        </td>
        <td><p>Show Signature Information in Web Browser</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_cosmic_sig_profile.html">show_cosmic_sig_profile()</a></code> </p>
        </td>
        <td><p>Plot Reference (Mainly COSMIC) Signature Profile</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_sig_consensusmap.html">show_sig_consensusmap()</a></code> </p>
        </td>
        <td><p>Show Signature Consensus Map</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-fit-and-bootstrap-visualization" class="hasAnchor"><a href="#section-fit-and-bootstrap-visualization" class="anchor"></a>Fit and bootstrap visualization</h2>
          <p class="section-desc"><p>Show signature fit and bootstrap analysis results.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="show_sig_fit.html">show_sig_fit()</a></code> </p>
        </td>
        <td><p>Show Signature Fit Result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_sig_bootstrap.html">show_sig_bootstrap_exposure()</a></code> <code><a href="show_sig_bootstrap.html">show_sig_bootstrap_error()</a></code> <code><a href="show_sig_bootstrap.html">show_sig_bootstrap_stability()</a></code> </p>
        </td>
        <td><p>Show Signature Bootstrap Analysis Results</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-output-analysis-results-to-local-machine" class="hasAnchor"><a href="#section-output-analysis-results-to-local-machine" class="anchor"></a>Output analysis results to local machine</h2>
          <p class="section-desc"><p>Save results and plots as files. For automatic analysis, please use <a href="https://github.com/ShixiangWang/sigminer.workflow">sigflow</a>.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="output_tally.html">output_tally()</a></code> </p>
        </td>
        <td><p>Output Tally Result in Barplots</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="output_sig.html">output_sig()</a></code> </p>
        </td>
        <td><p>Output Signature Results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="output_fit.html">output_fit()</a></code> </p>
        </td>
        <td><p>Output Signature Fitting Results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="output_bootstrap.html">output_bootstrap()</a></code> </p>
        </td>
        <td><p>Output Signature Bootstrap Fitting Results</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-signature-object-operation" class="hasAnchor"><a href="#section-signature-object-operation" class="anchor"></a>Signature object operation</h2>
          <p class="section-desc"><p>Obtain or modify <code>Signature</code> object information.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="sig_operation.html">sig_names()</a></code> <code><a href="sig_operation.html">sig_modify_names()</a></code> <code><a href="sig_operation.html">sig_number()</a></code> <code><a href="sig_operation.html">sig_attrs()</a></code> <code><a href="sig_operation.html">sig_signature()</a></code> <code><a href="sig_operation.html">sig_exposure()</a></code> </p>
        </td>
        <td><p>Obtain or Modify Signature Information</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-signature-analysis-and-corresponding-visualization" class="hasAnchor"><a href="#section-signature-analysis-and-corresponding-visualization" class="anchor"></a>Signature analysis and corresponding visualization</h2>
          <p class="section-desc"><p>Analyze signatures and output the result to visualization.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="get_sig_similarity.html">get_sig_similarity()</a></code> </p>
        </td>
        <td><p>Calculate Similarity between Identified Signatures and Reference Signatures</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_sig_rec_similarity.html">get_sig_rec_similarity()</a></code> </p>
        </td>
        <td><p>Get Reconstructed Profile Cosine Similarity, RSS, etc.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_sig_exposure.html">get_sig_exposure()</a></code> </p>
        </td>
        <td><p>Get Signature Exposure from 'Signature' Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_sig_feature_association.html">get_sig_feature_association()</a></code> </p>
        </td>
        <td><p>Calculate Association between Signature Exposures and Other Features</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_tidy_association.html">get_tidy_association()</a></code> </p>
        </td>
        <td><p>Get Tidy Signature Association Results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_cor.html">show_cor()</a></code> </p>
        </td>
        <td><p>A Simple and General Way for Association Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_sig_feature_corrplot.html">show_sig_feature_corrplot()</a></code> </p>
        </td>
        <td><p>Draw Corrplot for Signature Exposures and Other Features</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_groups.html">show_groups()</a></code> </p>
        </td>
        <td><p>Show Signature Contribution in Clusters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_groups.html">get_groups()</a></code> </p>
        </td>
        <td><p>Get Sample Groups from Signature Decomposition Information</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_group_comparison.html">get_group_comparison()</a></code> </p>
        </td>
        <td><p>Get Comparison Result between Signature Groups</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_group_comparison.html">show_group_comparison()</a></code> </p>
        </td>
        <td><p>Plot Group Comparison Result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="group_enrichment.html">group_enrichment()</a></code> </p>
        </td>
        <td><p>General Group Enrichment Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_group_enrichment.html">show_group_enrichment()</a></code> </p>
        </td>
        <td><p>Show Group Enrichment Result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_group_distribution.html">show_group_distribution()</a></code> </p>
        </td>
        <td><p>Show Groupped Variable Distribution</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_group_mapping.html">show_group_mapping()</a></code> </p>
        </td>
        <td><p>Map Groups using Sankey</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="enrich_component_strand_bias.html">enrich_component_strand_bias()</a></code> </p>
        </td>
        <td><p>Performs Strand Bias Enrichment Analysis for a Given Sample-by-Component Matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="simulation.html">simulate_signature()</a></code> <code><a href="simulation.html">simulate_catalogue()</a></code> <code><a href="simulation.html">simulate_catalogue_matrix()</a></code> </p>
        </td>
        <td><p>Simulation Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_shannon_diversity_index.html">get_shannon_diversity_index()</a></code> </p>
        </td>
        <td><p>Get Shannon Diversity Index for Signatures</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-package-datasets" class="hasAnchor"><a href="#section-package-datasets" class="anchor"></a>Package datasets</h2>
          <p class="section-desc"><p>Datasets used for signature identification and analysis (some stored in <code>extdata/</code>).</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="simulated_catalogs.html">simulated_catalogs</a></code> </p>
        </td>
        <td><p>A List of Simulated SBS-96 Catalog Matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CN.features.html">CN.features</a></code> </p>
        </td>
        <td><p>Classification Table of Copy Number Features Devised by Wang et al. for Method 'W'</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="centromeres.hg19.html">centromeres.hg19</a></code> </p>
        </td>
        <td><p>Location of Centromeres at Genome Build hg19</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="centromeres.hg38.html">centromeres.hg38</a></code> </p>
        </td>
        <td><p>Location of Centromeres at Genome Build hg38</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="centromeres.mm10.html">centromeres.mm10</a></code> </p>
        </td>
        <td><p>Location of Centromeres at Genome Build mm10</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="centromeres.mm9.html">centromeres.mm9</a></code> </p>
        </td>
        <td><p>Location of Centromeres at Genome Build mm9</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="chromsize.hg19.html">chromsize.hg19</a></code> </p>
        </td>
        <td><p>Chromosome Size of Genome Build hg19</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="chromsize.hg38.html">chromsize.hg38</a></code> </p>
        </td>
        <td><p>Chromosome Size of Genome Build hg38</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="chromsize.mm10.html">chromsize.mm10</a></code> </p>
        </td>
        <td><p>Chromosome Size of Genome Build mm10</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="chromsize.mm9.html">chromsize.mm9</a></code> </p>
        </td>
        <td><p>Chromosome Size of Genome Build mm9</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cytobands.hg19.html">cytobands.hg19</a></code> </p>
        </td>
        <td><p>Location of Chromosome Cytobands at Genome Build hg19</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cytobands.hg38.html">cytobands.hg38</a></code> </p>
        </td>
        <td><p>Location of Chromosome Cytobands at Genome Build hg38</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cytobands.mm10.html">cytobands.mm10</a></code> </p>
        </td>
        <td><p>Location of Chromosome Cytobands at Genome Build mm10</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cytobands.mm9.html">cytobands.mm9</a></code> </p>
        </td>
        <td><p>Location of Chromosome Cytobands at Genome Build mm9</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="transcript.hg19.html">transcript.hg19</a></code> </p>
        </td>
        <td><p>Merged Transcript Location at Genome Build hg19</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="transcript.hg38.html">transcript.hg38</a></code> </p>
        </td>
        <td><p>Merged Transcript Location at Genome Build hg38</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="transcript.mm10.html">transcript.mm10</a></code> </p>
        </td>
        <td><p>Merged Transcript Location at Genome Build mm10</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="transcript.mm9.html">transcript.mm9</a></code> </p>
        </td>
        <td><p>Merged Transcript Location at Genome Build mm9</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-helpers" class="hasAnchor"><a href="#section-helpers" class="anchor"></a>Helpers</h2>
          <p class="section-desc"><p>Helper functions used in the sigminer package.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="get_sig_db.html">get_sig_db()</a></code> </p>
        </td>
        <td><p>Get Curated Reference Signature Database</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sig_convert.html">sig_convert()</a></code> </p>
        </td>
        <td><p>Convert Signatures between different Genomic Distribution of Components</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_genome_annotation.html">get_genome_annotation()</a></code> </p>
        </td>
        <td><p>Get Genome Annotation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_bayesian_result.html">get_bayesian_result()</a></code> </p>
        </td>
        <td><p>Get Specified Bayesian NMF Result from Run</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_adj_p.html">get_adj_p()</a></code> </p>
        </td>
        <td><p>Get Adjust P Values from Group Comparison</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="use_color_style.html">use_color_style()</a></code> </p>
        </td>
        <td><p>Set Color Style for Plotting</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="add_h_arrow.html">add_h_arrow()</a></code> </p>
        </td>
        <td><p>Add Horizontal Arrow with Text Label to a ggplot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="add_labels.html">add_labels()</a></code> </p>
        </td>
        <td><p>Add Text Labels to a ggplot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cosine.html">cosine()</a></code> </p>
        </td>
        <td><p>Calculate Cosine Measures</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_cn_freq_table.html">get_cn_freq_table()</a></code> </p>
        </td>
        <td><p>Get CNV Frequency Table</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="same_size_clustering.html">same_size_clustering()</a></code> </p>
        </td>
        <td><p>Same Size Clustering</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_intersect_size.html">get_intersect_size()</a></code> </p>
        </td>
        <td><p>Get Overlap Size between Interval x and y</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-copy-number-analysis-and-visualization" class="hasAnchor"><a href="#section-copy-number-analysis-and-visualization" class="anchor"></a>Copy number analysis and visualization</h2>
          <p class="section-desc"><p>Functions for analyzing copy number data and visualization.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="transform_seg_table.html">transform_seg_table()</a></code> </p>
        </td>
        <td><p>Transform Copy Number Table</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_cn_ploidy.html">get_cn_ploidy()</a></code> </p>
        </td>
        <td><p>Get Ploidy from Absolute Copy Number Profile</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_Aneuploidy_score.html">get_Aneuploidy_score()</a></code> </p>
        </td>
        <td><p>Get Aneuploidy Score from Copy Number Profile</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scoring.html">scoring()</a></code> </p>
        </td>
        <td><p>Score Copy Number Profile</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_cn_profile.html">show_cn_profile()</a></code> </p>
        </td>
        <td><p>Show Sample Copy Number Profile</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_cn_circos.html">show_cn_circos()</a></code> </p>
        </td>
        <td><p>Show Copy Number Profile in Circos</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_cn_group_profile.html">show_cn_group_profile()</a></code> </p>
        </td>
        <td><p>Show Summary Copy Number Profile for Sample Groups</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_cn_freq_circos.html">show_cn_freq_circos()</a></code> </p>
        </td>
        <td><p>Show Copy Number Variation Frequency Profile with Circos</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_cn_distribution.html">show_cn_distribution()</a></code> </p>
        </td>
        <td><p>Show Copy Number Distribution either by Length or Chromosome</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_cn_features.html">show_cn_features()</a></code> </p>
        </td>
        <td><p>Show Copy Number Feature Distributions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show_cn_components.html">show_cn_components()</a></code> </p>
        </td>
        <td><p>Show Copy Number Components</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-extra-documentation" class="hasAnchor"><a href="#section-extra-documentation" class="anchor"></a>Extra documentation</h2>
          <p class="section-desc"><p>Extra function and introduction.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="hello.html">hello()</a></code> </p>
        </td>
        <td><p>Say Hello to Users</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sigminer.html">sigminer</a></code> </p>
        </td>
        <td><p>sigminer: Extract, Analyze and Visualize Signatures for Genomic Variations</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://orcid.org/0000-0001-9855-7357'>Shixiang Wang</a>, Ziyu Tao, Huimin Li, Tao Wu, <a href='https://orcid.org/0000-0002-7736-0077'>Xue-Song Liu</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


