<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Tally a Genomic Alteration Object — sig_tally • sigminer</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Tally a Genomic Alteration Object — sig_tally"><meta property="og:description" content="Tally a variation object like MAF, CopyNumber and return a matrix for NMF de-composition and more.
This is a generic function,
so it can be further extended to other mutation cases.
Please read details about how to set sex for identifying copy number signatures.
Please read https://osf.io/s93d5/ for the generation of SBS, DBS and ID (INDEL)
components."><meta property="og:image" content="https://shixiangwang.github.io/sigminer/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sigminer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.1.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/sigminer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ShixiangWang/sigminer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Tally a Genomic Alteration Object</h1>
    <small class="dont-index">Source: <a href="https://github.com/ShixiangWang/sigminer/blob/HEAD/R/sig_tally.R" class="external-link"><code>R/sig_tally.R</code></a></small>
    <div class="hidden name"><code>sig_tally.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Tally a variation object like <a href="MAF-class.html">MAF</a>, <a href="CopyNumber-class.html">CopyNumber</a> and return a matrix for NMF de-composition and more.
This is a generic function,
so it can be further extended to other mutation cases.
<strong>Please read details about how to set sex for identifying copy number signatures</strong>.
Please read <a href="https://osf.io/s93d5/" class="external-link">https://osf.io/s93d5/</a> for the generation of SBS, DBS and ID (INDEL)
components.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">sig_tally</span><span class="op">(</span><span class="va">object</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for CopyNumber</span>
<span class="fu">sig_tally</span><span class="op">(</span>
  <span class="va">object</span>,
  method <span class="op">=</span> <span class="st">"Wang"</span>,
  ignore_chrs <span class="op">=</span> <span class="cn">NULL</span>,
  indices <span class="op">=</span> <span class="cn">NULL</span>,
  add_loh <span class="op">=</span> <span class="cn">FALSE</span>,
  feature_setting <span class="op">=</span> <span class="fu">sigminer</span><span class="fu">::</span><span class="va"><a href="CN.features.html">CN.features</a></span>,
  cores <span class="op">=</span> <span class="fl">1</span>,
  keep_only_matrix <span class="op">=</span> <span class="cn">FALSE</span>,
  <span class="va">...</span>
<span class="op">)</span>

<span class="co"># S3 method for RS</span>
<span class="fu">sig_tally</span><span class="op">(</span><span class="va">object</span>, keep_only_matrix <span class="op">=</span> <span class="cn">FALSE</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for MAF</span>
<span class="fu">sig_tally</span><span class="op">(</span>
  <span class="va">object</span>,
  mode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SBS"</span>, <span class="st">"DBS"</span>, <span class="st">"ID"</span>, <span class="st">"ALL"</span><span class="op">)</span>,
  ref_genome <span class="op">=</span> <span class="st">"BSgenome.Hsapiens.UCSC.hg19"</span>,
  genome_build <span class="op">=</span> <span class="cn">NULL</span>,
  add_trans_bias <span class="op">=</span> <span class="cn">FALSE</span>,
  ignore_chrs <span class="op">=</span> <span class="cn">NULL</span>,
  use_syn <span class="op">=</span> <span class="cn">TRUE</span>,
  keep_only_matrix <span class="op">=</span> <span class="cn">FALSE</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>object</dt>
<dd><p>a <a href="CopyNumber-class.html">CopyNumber</a> object or <a href="MAF-class.html">MAF</a> object or SV object (from <a href="read_sv_as_rs.html">read_sv_as_rs</a>).</p></dd>
<dt>...</dt>
<dd><p>custom setting for operating object. Detail see S3 method for
corresponding class (e.g. <code>CopyNumber</code>).</p></dd>
<dt>method</dt>
<dd><p>method for feature classification, can be one of
"Wang" ("W"), "S" (for method described in Steele et al. 2019).</p></dd>
<dt>ignore_chrs</dt>
<dd><p>Chromsomes to ignore from analysis. e.g. chrX and chrY.</p></dd>
<dt>indices</dt>
<dd><p>integer vector indicating segments to keep.</p></dd>
<dt>add_loh</dt>
<dd><p>flag to add LOH classifications.</p></dd>
<dt>feature_setting</dt>
<dd><p>a <code>data.frame</code> used for classification.
<strong>Only used when method is "Wang" ("W")</strong>.
Default is <a href="CN.features.html">CN.features</a>. Users can also set custom input with "feature",
"min" and "max" columns available. Valid features can be printed by
<code>unique(CN.features$feature)</code>.</p></dd>
<dt>cores</dt>
<dd><p>number of computer cores to run this task.
You can use <code><a href="https://future.futureverse.org/reference/re-exports.html" class="external-link">future::availableCores()</a></code> function to check how
many cores you can use.</p></dd>
<dt>keep_only_matrix</dt>
<dd><p>if <code>TRUE</code>, keep only matrix for signature extraction.
For a <code>MAF</code> object, this will just return the most useful matrix.</p></dd>
<dt>mode</dt>
<dd><p>type of mutation matrix to extract, can be one of 'SBS', 'DBS' and 'ID'.</p></dd>
<dt>ref_genome</dt>
<dd><p>'BSgenome.Hsapiens.UCSC.hg19', 'BSgenome.Hsapiens.UCSC.hg38',
'BSgenome.Mmusculus.UCSC.mm10',  'BSgenome.Mmusculus.UCSC.mm9', etc.</p></dd>
<dt>genome_build</dt>
<dd><p>genome build 'hg19', 'hg38', 'mm9' or "mm10", if not set, guess it by <code>ref_genome</code>.</p></dd>
<dt>add_trans_bias</dt>
<dd><p>if <code>TRUE</code>, consider transcriptional bias categories.
'T:' for Transcribed (the variant is on the transcribed strand);
'U:' for Un-transcribed (the variant is on the untranscribed strand);
'B:' for Bi-directional (the variant is on both strand and is transcribed either way);
'N:' for Non-transcribed (the variant is in a non-coding region and is untranslated);
'Q:' for Questionable.
<strong>NOTE</strong>: the result counts of 'B' and 'N' labels are a little different from
SigProfilerMatrixGenerator, the reason is unknown (may be caused by annotation file).</p></dd>
<dt>use_syn</dt>
<dd><p>Logical. If <code>TRUE</code>, include synonymous variants in analysis.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>a <code>list</code> contains a <code>matrix</code> used for NMF de-composition.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>For identifying copy number signatures, we have to derive copy number
features firstly. Due to the difference of copy number values in sex chromosomes
between male and female, we have to do an extra step <strong>if we don't want to
ignore them</strong>.</p>
<p>I create two options to control this, the default values are shown as
the following, you can use the same way to set (per R session).</p>
<p><code>options(sigminer.sex = "female", sigminer.copynumber.max = NA_integer_)</code></p><ul><li><p>If your cohort are all females, you can totally ignore this.</p></li>
<li><p>If your cohort are all males, set <code>sigminer.sex</code> to 'male' and
<code>sigminer.copynumber.max</code> to a proper value (the best is consistent
with <a href="read_copynumber.html">read_copynumber</a>).</p></li>
<li><p>If your cohort contains both males and females, set <code>sigminer.sex</code>
as a <code>data.frame</code> with two columns "sample" and "sex". And
set <code>sigminer.copynumber.max</code> to a proper value (the best is consistent
with <a href="read_copynumber.html">read_copynumber</a>).</p></li>
</ul></div>
    <div id="methods-by-class-">
    <h2>Methods (by class)</h2>
    
<ul><li><p><code>CopyNumber</code>: Returns copy number features, components and component-by-sample matrix</p></li>
<li><p><code>RS</code>: Returns genome rearrangement sample-by-component matrix</p></li>
<li><p><code>MAF</code>: Returns SBS mutation sample-by-component matrix and APOBEC enrichment</p></li>
</ul></div>
    <div id="references">
    <h2>References</h2>
    <p>Wang, Shixiang, et al. "Copy number signature analyses in prostate cancer reveal
distinct etiologies and clinical outcomes." medRxiv (2020).</p>
<p>Steele, Christopher D., et al. "Undifferentiated sarcomas develop through
distinct evolutionary pathways." Cancer Cell 35.3 (2019): 441-456.</p>
<p>Mayakonda, Anand, et al. "Maftools: efficient and comprehensive analysis of somatic variants in cancer." Genome research 28.11 (2018): 1747-1756.</p>
<p>Roberts SA, Lawrence MS, Klimczak LJ, et al. An APOBEC Cytidine Deaminase Mutagenesis Pattern is Widespread in Human Cancers. Nature genetics. 2013;45(9):970-976. doi:10.1038/ng.2702.</p>
<p>Bergstrom EN, Huang MN, Mahto U, Barnes M, Stratton MR, Rozen SG, Alexandrov LB: SigProfilerMatrixGenerator: a tool for visualizing and exploring patterns of small mutational events. BMC Genomics 2019, 20:685 https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6041-2</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="sig_estimate.html">sig_estimate</a> for estimating signature number for <a href="sig_extract.html">sig_extract</a>,
<a href="sig_auto_extract.html">sig_auto_extract</a> for extracting signatures using automatic relevance determination technique.</p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Shixiang Wang</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="co"># Load copy number object</span>
<span class="fu"><a href="https://rdrr.io/pkg/R.oo/man/Non-documented_objects.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"toy_copynumber.RData"</span>,
  package <span class="op">=</span> <span class="st">"sigminer"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span><span class="op">)</span>
<span class="co"># \donttest{</span>
<span class="co"># Use method designed by Wang, Shixiang et al.</span>
<span class="va">cn_tally_W</span> <span class="op">&lt;-</span> <span class="fu">sig_tally</span><span class="op">(</span><span class="va">cn</span>, method <span class="op">=</span> <span class="st">"W"</span><span class="op">)</span>
<span class="co"># }</span>
<span class="co"># Use method designed by Steele et al.</span>
<span class="co"># See example in read_copynumber</span>
<span class="co"># \donttest{</span>
<span class="co"># Prepare SBS signature analysis</span>
<span class="va">laml.maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"tcga_laml.maf.gz"</span>, package <span class="op">=</span> <span class="st">"maftools"</span><span class="op">)</span>
<span class="va">laml</span> <span class="op">&lt;-</span> <span class="fu"><a href="read_maf.html">read_maf</a></span><span class="op">(</span>maf <span class="op">=</span> <span class="va">laml.maf</span><span class="op">)</span>
<span class="kw">if</span> <span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"BSgenome.Hsapiens.UCSC.hg19"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">mt_tally</span> <span class="op">&lt;-</span> <span class="fu">sig_tally</span><span class="op">(</span>
    <span class="va">laml</span>,
    ref_genome <span class="op">=</span> <span class="st">"BSgenome.Hsapiens.UCSC.hg19"</span>,
    use_syn <span class="op">=</span> <span class="cn">TRUE</span>
  <span class="op">)</span>
  <span class="va">mt_tally</span><span class="op">$</span><span class="va">nmf_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>

  <span class="co">## Use strand bias categories</span>
  <span class="va">mt_tally</span> <span class="op">&lt;-</span> <span class="fu">sig_tally</span><span class="op">(</span>
    <span class="va">laml</span>,
    ref_genome <span class="op">=</span> <span class="st">"BSgenome.Hsapiens.UCSC.hg19"</span>,
    use_syn <span class="op">=</span> <span class="cn">TRUE</span>, add_trans_bias <span class="op">=</span> <span class="cn">TRUE</span>
  <span class="op">)</span>
  <span class="co">## Test it by enrichment analysis</span>
  <span class="fu"><a href="enrich_component_strand_bias.html">enrich_component_strand_bias</a></span><span class="op">(</span><span class="va">mt_tally</span><span class="op">$</span><span class="va">nmf_matrix</span><span class="op">)</span>
  <span class="fu"><a href="enrich_component_strand_bias.html">enrich_component_strand_bias</a></span><span class="op">(</span><span class="va">mt_tally</span><span class="op">$</span><span class="va">all_matrices</span><span class="op">$</span><span class="va">SBS_24</span><span class="op">)</span>
<span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"Please install package 'BSgenome.Hsapiens.UCSC.hg19' firstly!"</span><span class="op">)</span>
<span class="op">}</span>
<span class="co"># }</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://orcid.org/0000-0001-9855-7357" class="external-link">Shixiang
Wang</a>, Ziyu Tao, Huimin Li, Tao Wu,
<a href="https://orcid.org/0000-0002-7736-0077" class="external-link">Xue-Song Liu</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

