<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Obtain Bootstrap Distribution of Signature Exposures of a Certain Tumor Sample — sig_fit_bootstrap • sigminer</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Obtain Bootstrap Distribution of Signature Exposures of a Certain Tumor Sample — sig_fit_bootstrap" />
<meta property="og:description" content="This can be used to obtain the confidence of signature exposures or search
the suboptimal decomposition solution." />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sigminer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/sigminer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ShixiangWang/sigminer/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Obtain Bootstrap Distribution of Signature Exposures of a Certain Tumor Sample</h1>
    <small class="dont-index">Source: <a href='https://github.com/ShixiangWang/sigminer/blob/master/R/sig_fit_bootstrap.R'><code>R/sig_fit_bootstrap.R</code></a></small>
    <div class="hidden name"><code>sig_fit_bootstrap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This can be used to obtain the confidence of signature exposures or search
the suboptimal decomposition solution.</p>
    </div>

    <pre class="usage"><span class='fu'>sig_fit_bootstrap</span><span class='op'>(</span>
  <span class='va'>catalog</span>,
  <span class='va'>sig</span>,
  n <span class='op'>=</span> <span class='fl'>100L</span>,
  sig_index <span class='op'>=</span> <span class='cn'>NULL</span>,
  sig_db <span class='op'>=</span> <span class='st'>"legacy"</span>,
  db_type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>""</span>, <span class='st'>"human-exome"</span>, <span class='st'>"human-genome"</span><span class='op'>)</span>,
  show_index <span class='op'>=</span> <span class='cn'>TRUE</span>,
  method <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"QP"</span>, <span class='st'>"NNLS"</span>, <span class='st'>"SA"</span><span class='op'>)</span>,
  auto_reduce <span class='op'>=</span> <span class='cn'>FALSE</span>,
  SA_not_bootstrap <span class='op'>=</span> <span class='cn'>FALSE</span>,
  type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"absolute"</span>, <span class='st'>"relative"</span><span class='op'>)</span>,
  rel_threshold <span class='op'>=</span> <span class='fl'>0</span>,
  mode <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"SBS"</span>, <span class='st'>"DBS"</span>, <span class='st'>"ID"</span>, <span class='st'>"copynumber"</span><span class='op'>)</span>,
  find_suboptimal <span class='op'>=</span> <span class='cn'>FALSE</span>,
  suboptimal_ref_error <span class='op'>=</span> <span class='cn'>NULL</span>,
  suboptimal_factor <span class='op'>=</span> <span class='fl'>1.05</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>catalog</th>
      <td><p>a named numeric vector or a numeric matrix with dimension Nx1.
N is the number of component, 1 is the sample.</p></td>
    </tr>
    <tr>
      <th>sig</th>
      <td><p>a <code>Signature</code> object obtained either from <a href='sig_extract.html'>sig_extract</a> or <a href='sig_auto_extract.html'>sig_auto_extract</a>,
or just a raw signature matrix/<code>data.frame</code> with row representing components (motifs) and
column representing signatures.</p></td>
    </tr>
    <tr>
      <th>n</th>
      <td><p>the number of bootstrap replicates.</p></td>
    </tr>
    <tr>
      <th>sig_index</th>
      <td><p>a vector for signature index. "ALL" for all signatures.</p></td>
    </tr>
    <tr>
      <th>sig_db</th>
      <td><p>default 'legacy', it can be 'legacy' (for <a href='https://cancer.sanger.ac.uk/cosmic/signatures_v2.tt/'>COSMIC v2 'SBS'</a>),
'SBS', 'DBS', 'ID' and 'TSB' (for <a href='https://cancer.sanger.ac.uk/cosmic/signatures/'>COSMIV v3.1 signatures</a>).
For more specific details, it can also be 'SBS_hg19', 'SBS_hg38',
'SBS_mm9', 'SBS_mm10', 'DBS_hg19', 'DBS_hg38', 'DBS_mm9', 'DBS_mm10' to use
COSMIC v3 reference signatures from Alexandrov, Ludmil B., et al. (2020) (reference #1).
In addition, it can be one of "SBS_Nik_lab_Organ", "RS_Nik_lab_Organ",
"SBS_Nik_lab", "RS_Nik_lab" to refer reference signatures from
Degasperi, Andrea, et al. (2020) (reference #2).
<strong>Note</strong>: the signature profile for different genome builds are basically same.
And specific database (e.g. 'SBS_mm10') contains less signatures than all COSMIC
signatures (because some signatures are not detected from Alexandrov, Ludmil B., et al. (2020)).</p></td>
    </tr>
    <tr>
      <th>db_type</th>
      <td><p>only used when <code>sig_db</code> is enabled.
"" for keeping default, "human-exome" for transforming to exome frequency of component,
and "human-genome" for transforming to whole genome frequency of component.
Currently only works for 'SBS'.</p></td>
    </tr>
    <tr>
      <th>show_index</th>
      <td><p>if <code>TRUE</code>, show valid indices.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>method to solve the minimazation problem.
'NNLS' for non-negative least square; 'QP' for quadratic programming; 'SA' for simulated annealing.</p></td>
    </tr>
    <tr>
      <th>auto_reduce</th>
      <td><p>if <code>TRUE</code>, try reducing the input reference signatures to increase
the cosine similarity of reconstructed profile to observed profile.</p></td>
    </tr>
    <tr>
      <th>SA_not_bootstrap</th>
      <td><p>if <code>TRUE</code>, directly run 'SA' multiple times with original input instead of
bootstrap samples.</p></td>
    </tr>
    <tr>
      <th>type</th>
      <td><p>'absolute' for signature exposure and 'relative' for signature relative exposure.</p></td>
    </tr>
    <tr>
      <th>rel_threshold</th>
      <td><p>numeric vector, a signature with relative exposure
lower than (equal is included, i.e. <code>&lt;=</code>) this value will be set to 0
(both absolute exposure and relative exposure).
In this case, sum of signature contribution may not equal to 1.</p></td>
    </tr>
    <tr>
      <th>mode</th>
      <td><p>signature type for plotting, now supports 'copynumber', 'SBS',
'DBS', 'ID' and 'RS' (genome rearrangement signature).</p></td>
    </tr>
    <tr>
      <th>find_suboptimal</th>
      <td><p>logical, if <code>TRUE</code>, find suboptimal decomposition with
slightly higher error than the optimal solution by method 'SA'. This is useful
to explore hidden dependencies between signatures. More see reference.</p></td>
    </tr>
    <tr>
      <th>suboptimal_ref_error</th>
      <td><p>baseline error used for finding suboptimal solution.
if it is <code>NULL</code>, then use 'SA' method to obtain the optimal error.</p></td>
    </tr>
    <tr>
      <th>suboptimal_factor</th>
      <td><p>suboptimal factor to get suboptimal error, default is <code>1.05</code>,
i.e., suboptimal error is <code>1.05</code> times baseline error.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>control parameters passing to argument <code>control</code> in <code>GenSA</code> function when use method 'SA'.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a <code>list</code></p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Huang X, Wojtowicz D, Przytycka TM. Detecting presence of mutational signatures in cancer with confidence. Bioinformatics. 2018;34(2):330–337. doi:10.1093/bioinformatics/btx604</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='report_bootstrap_p_value.html'>report_bootstrap_p_value</a>, <a href='sig_fit.html'>sig_fit</a>, <a href='sig_fit_bootstrap_batch.html'>sig_fit_bootstrap_batch</a></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='va'>W</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>matrix</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>2</span>, <span class='fl'>3</span>, <span class='fl'>4</span>, <span class='fl'>5</span>, <span class='fl'>6</span><span class='op'>)</span>, ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>W</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"sig1"</span>, <span class='st'>"sig2"</span><span class='op'>)</span>
<span class='va'>W</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/apply.html'>apply</a></span><span class='op'>(</span><span class='va'>W</span>, <span class='fl'>2</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='va'>x</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>H</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>matrix</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>2</span>, <span class='fl'>5</span>, <span class='fl'>3</span>, <span class='fl'>6</span>, <span class='fl'>1</span>, <span class='fl'>9</span>, <span class='fl'>1</span>, <span class='fl'>2</span><span class='op'>)</span>, ncol <span class='op'>=</span> <span class='fl'>4</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>H</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"samp"</span>, <span class='fl'>1</span><span class='op'>:</span><span class='fl'>4</span><span class='op'>)</span>

<span class='va'>V</span> <span class='op'>&lt;-</span> <span class='va'>W</span> <span class='op'>%*%</span> <span class='va'>H</span>
<span class='va'>V</span>

<span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/ns-load.html'>requireNamespace</a></span><span class='op'>(</span><span class='st'>"quadprog"</span>, quietly <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>H_bootstrap</span> <span class='op'>&lt;-</span> <span class='fu'>sig_fit_bootstrap</span><span class='op'>(</span><span class='va'>V</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>, <span class='va'>W</span>, n <span class='op'>=</span> <span class='fl'>10</span>, type <span class='op'>=</span> <span class='st'>"absolute"</span><span class='op'>)</span>
  <span class='co'>## Typically, you have to run many times to get close to the answer</span>
  <span class='fu'><a href='https://rdrr.io/r/graphics/boxplot.html'>boxplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>H_bootstrap</span><span class='op'>$</span><span class='va'>expo</span><span class='op'>)</span><span class='op'>)</span>
  <span class='va'>H</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span>

  <span class='co'>## Return P values</span>
  <span class='co'>## In practice, run times &gt;= 100</span>
  <span class='co'>## is recommended</span>
  <span class='fu'><a href='report_bootstrap_p_value.html'>report_bootstrap_p_value</a></span><span class='op'>(</span><span class='va'>H_bootstrap</span><span class='op'>)</span>
  <span class='co'>## For multiple samples</span>
  <span class='co'>## Input a list</span>
  <span class='fu'><a href='report_bootstrap_p_value.html'>report_bootstrap_p_value</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>samp1 <span class='op'>=</span> <span class='va'>H_bootstrap</span>, samp2 <span class='op'>=</span> <span class='va'>H_bootstrap</span><span class='op'>)</span><span class='op'>)</span>

  <span class='co'>#   ## Find suboptimal decomposition</span>
  <span class='co'>#   H_suboptimal &lt;- sig_fit_bootstrap(V[, 1], W,</span>
  <span class='co'>#     n = 10,</span>
  <span class='co'>#     type = "absolute",</span>
  <span class='co'>#     method = "SA",</span>
  <span class='co'>#     find_suboptimal = TRUE</span>
  <span class='co'>#   )</span>
<span class='op'>}</span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://orcid.org/0000-0001-9855-7357'>Shixiang Wang</a>, Ziyu Tao, Huimin Li, Tao Wu, <a href='https://orcid.org/0000-0002-7736-0077'>Xue-Song Liu</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


